{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-06T15:08:19.635893Z",
     "start_time": "2018-04-06T15:11:07.910Z"
    },
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "<p style='text-align: justify;'> Analysis to explore and discover genes and non-coding region under positive selection through 1000GP phase 3 data obtained and processed by [PopHuman](https://pophuman.uab.cat) database. Statistic showed below were retrieve from 10-kb windows and 100-kb windows. Statistical results were obtained using 10-kb windows data. 100-kb windows were used to explore and describe chromosome and populations distributions.\n",
    "\n",
    "The pipeline requires [R kernel](https://irkernel.github.io/) for jupyter and use a maximum of 20GB of RAM approximately.</p>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "Loading R libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-29T09:31:24.438993Z",
     "start_time": "2018-05-29T09:35:03.481Z"
    }
   },
   "outputs": [],
   "source": [
    "library(\"data.table\")\n",
    "library(\"dplyr\")\n",
    "library(\"stringr\")\n",
    "library(\"purrr\")\n",
    "library(\"intervals\")\n",
    "library(\"ggplot2\")\n",
    "library(\"cowplot\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Open and merge bedGraph tables in order to extract and create a data.frame with equal rows between populations and test through unique coordinates"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "<p style='text-align: justify;'>We compile uniques start and end coordinates in each test and population to establish unique ID to each 10-kb window. The ID is first establish using [data.table](https://github.com/Rdatatable/data.table/wiki) library and subsequently merge, creating a data.frame with unique IDs associate to populations and tests. If a population have not calculated a statistic in a determinated window value will be **NULL**</p>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### Associate windows, statisitics and populations in a data.frame"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "#### 10-kb windows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-29T14:15:00.823983Z",
     "start_time": "2018-05-29T14:16:53.671Z"
    },
    "code_folding": [
     5
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dfTemporal <- NULL\n",
    "windows <- NULL\n",
    "\n",
    "pops <- c(\"ACB\", \"ASW\", \"BEB\", \"CDX\", \"CEU\", \"CHB\", \"CHS\", \"CLM\", \"ESN\", \"FIN\", \"GBR\", \n",
    "    \"GIH\", \"GWD\", \"IBS\", \"ITU\", \"JPT\", \"KHV\", \"LWK\", \"MSL\", \"MXL\", \"PEL\", \"PJL\", \n",
    "    \"PUR\", \"STU\", \"TSI\", \"YRI\")\n",
    "stat <- c(\"Tajima_D\", \"FayWu_H\", \"FuLi_D\", \"FuLi_F\", \"iHS\", \"S\", \"theta\", \"Pi\")\n",
    "for (j in pops) {\n",
    "    print(\"+++++++++++++++++++++\")\n",
    "    print(j)\n",
    "    print(\"+++++++++++++++++++++\")\n",
    "    for (i in stat) {\n",
    "        folder <- paste0(\"~/pophuman.uab.cat/Data/bedGraph/10kb/\", i, \"/\", i, \"_\", \n",
    "            j, \"_10kb.bedGraph\")\n",
    "        print(i)\n",
    "        df <- fread(folder, header = F, sep = \"\\t\")\n",
    "        colnames(df) <- c(\"chr\", \"startW\", \"endW\", paste0(i))\n",
    "        tempVar <- noquote(paste0(i))\n",
    "        df[[tempVar]] <- df[[i]]\n",
    "        df[[\"pop\"]] <- j\n",
    "        if (i == \"Tajima_D\") {\n",
    "            # Tajima_D first iteration\n",
    "            dfTemporal <- df\n",
    "        } else {\n",
    "            dfTemporal <- merge(dfTemporal, df, by = c(\"chr\", \"startW\", \"endW\", \"pop\"), \n",
    "                all = T)\n",
    "        }\n",
    "    }\n",
    "    windows <- rbind(windows, dfTemporal)\n",
    "}\n",
    "\n",
    "windows[[\"Fst\"]] <- NA\n",
    "windows[[\"XPEHH\"]] <- NA\n",
    "windows[[\"chr\"]] <- factor(windows[[\"chr\"]], levels = c(\"chr1\", \"chr2\", \"chr3\", \"chr4\", \n",
    "    \"chr5\", \"chr6\", \"chr7\", \"chr8\", \"chr9\", \"chr10\", \"chr11\", \"chr12\", \"chr13\", \"chr14\", \n",
    "    \"chr15\", \"chr16\", \"chr17\", \"chr18\", \"chr19\", \"chr20\", \"chr21\", \"chr22\", \"chrX\", \n",
    "    \"chrY\"))\n",
    "windows <- windows[order(windows[[\"chr\"]], windows[[\"startW\"]])]\n",
    "windows[, `:=`(ID, .GRP), by = .(chr, startW, endW)]\n",
    "\n",
    "windows[[\"chr\"]] <- as.character(windows[[\"chr\"]])\n",
    "\n",
    "save(windows, file = \"~/pophuman.uab.cat/Results/10kb/windowsStatisticsPopulations.RData\")\n",
    "fwrite(windows, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/windowsStatisticsPopulations.tab\", \n",
    "    quote = F, col.names = T, row.names = F, sep = \"\\t\")\n",
    "head(windows)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "#### 100-kb windows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-23T14:26:02.946971Z",
     "start_time": "2018-05-23T14:28:25.638Z"
    },
    "code_folding": [
     5
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "dfTemporal <- NULL\n",
    "windows100kb <- NULL\n",
    "\n",
    "pops <- c(\"ACB\", \"ASW\", \"BEB\", \"CDX\", \"CEU\", \"CHB\", \"CHS\", \"CLM\", \"ESN\", \"FIN\", \"GBR\", \n",
    "    \"GIH\", \"GWD\", \"IBS\", \"ITU\", \"JPT\", \"KHV\", \"LWK\", \"MSL\", \"MXL\", \"PEL\", \"PJL\", \n",
    "    \"PUR\", \"STU\", \"TSI\", \"YRI\")\n",
    "stat <- c(\"Tajima_D\", \"FayWu_H\", \"FuLi_D\", \"FuLi_F\", \"S\", \"theta\", \"Pi\")\n",
    "for (j in pops) {\n",
    "    print(\"+++++++++++++++++++++\")\n",
    "    print(j)\n",
    "    print(\"+++++++++++++++++++++\")\n",
    "    for (i in stat) {\n",
    "        folder <- paste0(\"~/pophuman.uab.cat/Data/bedGraph/100kb/\", i, \"/\", i, \"_\", \n",
    "            j, \"_100kb.bedGraph\")\n",
    "        print(i)\n",
    "        df <- fread(folder, header = F, sep = \"\\t\")\n",
    "        colnames(df) <- c(\"chr\", \"startW\", \"endW\", paste0(i))\n",
    "        tempVar <- noquote(paste0(i))\n",
    "        df[[tempVar]] <- df[[i]]\n",
    "        df[[\"pop\"]] <- j\n",
    "        if (i == \"Tajima_D\") {\n",
    "            # Tajima_D first iteration\n",
    "            dfTemporal <- df\n",
    "        } else {\n",
    "            dfTemporal <- merge(dfTemporal, df, by = c(\"chr\", \"startW\", \"endW\", \"pop\"), \n",
    "                all = T)\n",
    "        }\n",
    "    }\n",
    "    windows100kb <- rbind(windows100kb, dfTemporal)\n",
    "}\n",
    "\n",
    "windows100kb[[\"Fst\"]] <- NA\n",
    "windows100kb[[\"XPEHH\"]] <- NA\n",
    "windows100kb[[\"chr\"]] <- factor(windows100kb[[\"chr\"]], levels = c(\"chr1\", \"chr2\", \n",
    "    \"chr3\", \"chr4\", \"chr5\", \"chr6\", \"chr7\", \"chr8\", \"chr9\", \"chr10\", \"chr11\", \"chr12\", \n",
    "    \"chr13\", \"chr14\", \"chr15\", \"chr16\", \"chr17\", \"chr18\", \"chr19\", \"chr20\", \"chr21\", \n",
    "    \"chr22\", \"chrX\", \"chrY\"))\n",
    "windows100kb <- windows100kb[order(windows100kb[[\"chr\"]], windows100kb[[\"startW\"]])]\n",
    "windows100kb[, `:=`(ID, .GRP), by = .(chr, startW, endW)]\n",
    "windows100kb[[\"chr\"]] <- as.character(windows100kb[[\"chr\"]])\n",
    "\n",
    "save(windows100kb, file = \"~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData\")\n",
    "fwrite(windows100kb, file = \"~/pophuman.uab.cat/Results/100kb/tabFiles/windowsStatisticsPopulations100kb.tab\", \n",
    "    quote = F, col.names = T, row.names = F, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "#### Calculate mean Fst by windows. Add Fst and XPEHH to windows, statisitics and populations data.frame"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "##### 10-kb windows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-29T14:50:37.926634Z",
     "start_time": "2018-05-29T14:22:02.640Z"
    },
    "code_folding": [
     0
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if (!exists(\"windows\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/windowsStatisticsPopulations.RData\")\n",
    "}\n",
    "\n",
    "windowsToStat <- unique(windows[, c(\"chr\", \"startW\", \"endW\", \"ID\")])  #Extract only chr, startWindow, endWindow and ID\n",
    "\n",
    "# FST\n",
    "dfTemporal <- NULL\n",
    "pops <- c(\"CEU2CHB\", \"CEU2YRI\", \"YRI2CHB\")\n",
    "chr <- c(1:22, \"X\", \"Y\")\n",
    "for (i in pops) {\n",
    "    if (i == \"CEU2CHB\") {\n",
    "        windowsToStat <- windows[windows[[\"pop\"]] == \"CEU\" | windows[[\"pop\"]] == \n",
    "            \"CHB\", c(\"chr\", \"startW\", \"endW\", \"ID\", \"pop\")]\n",
    "        windowsToStat <- unique(windowsToStat[duplicated(windowsToStat[, \"ID\"]), \n",
    "            c(\"chr\", \"startW\", \"endW\", \"ID\")])\n",
    "        groupA <- windowsToStat\n",
    "        groupA <- unique(groupA)\n",
    "        setkey(groupA, chr, startW, endW)\n",
    "        groupA <- na.omit(groupA)\n",
    "    } else if (i == \"CEU2YRI\") {\n",
    "        windowsToStat <- windows[windows[[\"pop\"]] == \"CEU\" | windows[[\"pop\"]] == \n",
    "            \"YRI\", c(\"chr\", \"startW\", \"endW\", \"ID\", \"pop\")]\n",
    "        windowsToStat <- unique(windowsToStat[duplicated(windowsToStat[, \"ID\"]), \n",
    "            c(\"chr\", \"startW\", \"endW\", \"ID\")])\n",
    "        groupA <- as.data.table(windowsToStat)\n",
    "        groupA <- unique(groupA)\n",
    "        setkey(groupA, chr, startW, endW)\n",
    "        groupA <- na.omit(groupA)\n",
    "    } else if (i == \"YRI2CHB\") {\n",
    "        windowsToStat <- windows[windows[[\"pop\"]] == \"YRI\" | windows[[\"pop\"]] == \n",
    "            \"CHB\", c(\"chr\", \"startW\", \"endW\", \"ID\", \"pop\")]\n",
    "        windowsToStat <- unique(windowsToStat[duplicated(windowsToStat[, \"ID\"]), \n",
    "            c(\"chr\", \"startW\", \"endW\", \"ID\")])\n",
    "        groupA <- as.data.table(windowsToStat)\n",
    "        groupA <- unique(groupA)\n",
    "        setkey(groupA, chr, startW, endW)\n",
    "        groupA <- na.omit(groupA)\n",
    "    }\n",
    "    print(\"+++++++++++++++++\")\n",
    "    print(i)\n",
    "    print(\"+++++++++++++++++\")\n",
    "    for (j in chr) {\n",
    "        print(j)\n",
    "        folder <- paste0(\"~/pophuman.uab.cat/Data/Fst/\", j, \"/\", i, \"chr\", j, \".weir.fst\")\n",
    "        df <- fread(folder, header = T, sep = \"\\t\")\n",
    "        colnames(df) <- c(\"chr\", \"start\", \"Fst\")\n",
    "        df[[\"start\"]] <- df[[\"start\"]]\n",
    "        df[[\"end\"]] <- df[[\"start\"]]\n",
    "        df[[\"chr\"]] <- paste0(\"chr\", j)\n",
    "        df[[\"pop\"]] <- i\n",
    "        \n",
    "        groupB <- df\n",
    "        setkey(groupB, chr, start, end)\n",
    "        \n",
    "        over <- foverlaps(groupA, groupB, nomatch = 0)\n",
    "        over <- unique(over)\n",
    "        \n",
    "        dfTemporal <- rbind(dfTemporal, over)\n",
    "        \n",
    "    }\n",
    "}\n",
    "\n",
    "fst <- dfTemporal %>% group_by(ID, pop, startW, endW, chr) %>% summarise(Fst = mean(na.omit(Fst)))\n",
    "fst <- as.data.table(fst)\n",
    "\n",
    "# XPEHH\n",
    "pops <- c(\"CHB2CEU\", \"CHB2YRI\", \"CEU2YRI\")\n",
    "stat <- c(\"XPEHH\")\n",
    "dfTemporal <- NULL\n",
    "df <- NULL\n",
    "\n",
    "for (j in pops) {\n",
    "    print(\"+++++++++++++++++++++\")\n",
    "    print(j)\n",
    "    print(\"+++++++++++++++++++++\")\n",
    "    df <- NULL\n",
    "    for (i in stat) {\n",
    "        folder <- paste0(\"~/pophuman.uab.cat/Data/bedGraph/10kb/\", i, \"/\", i, \"_\", \n",
    "            j, \"_10kb.bedGraph\")\n",
    "        print(i)\n",
    "        df <- fread(folder, header = F, sep = \"\\t\")\n",
    "        colnames(df) <- c(\"chr\", \"startW\", \"endW\", paste0(i))\n",
    "        tempVar <- noquote(paste0(i))\n",
    "        df[[tempVar]] <- df[[i]]\n",
    "        if (j == \"CHB2CEU\") {\n",
    "            j <- \"CEU2CHB\"\n",
    "        } else if (j == \"CHB2YRI\") {\n",
    "            j <- \"YRI2CHB\"\n",
    "        }\n",
    "        \n",
    "        df[[\"pop\"]] <- j\n",
    "        \n",
    "    }\n",
    "    dfTemporal <- rbind(dfTemporal, df)\n",
    "}\n",
    "\n",
    "windowsID <- unique(windows[, c(\"chr\", \"startW\", \"endW\", \"ID\")])\n",
    "\n",
    "XPEHH <- merge(windowsID, dfTemporal, by = c(\"chr\", \"startW\", \"endW\"), all = T)\n",
    "fst.XPEHH <- merge(fst, XPEHH, by = c(\"ID\", \"pop\", \"chr\", \"startW\", \"endW\"), all = T)\n",
    "windows <- merge(as.data.frame(fst.XPEHH), as.data.frame(windows), by = c(\"pop\", \n",
    "    \"chr\", \"startW\", \"endW\", \"Fst\", \"XPEHH\", \"ID\"), all = T)\n",
    "\n",
    "windows <- as.data.table(windows)\n",
    "windows <- windows[order(windows[[\"chr\"]], windows[[\"startW\"]])]\n",
    "windows <- windows[!is.na(windows[[\"pop\"]]), ]\n",
    "# Overwrite existing data.frames\n",
    "save(windows, file = \"~/pophuman.uab.cat/Results/10kb/windowsStatisticsPopulations.RData\")\n",
    "fwrite(windows, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/windowsStatisticsPopulations.tab\", \n",
    "    quote = F, col.names = T, row.names = F, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "##### 100-kb windows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-05-23T14:29:16.335Z"
    },
    "code_folding": [
     12,
     13,
     21,
     28
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if (!exists(\"windows100kb\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData\")\n",
    "}\n",
    "\n",
    "windowsToStat <- unique(windows100kb[, c(\"chr\", \"startW\", \"endW\", \"ID\")])  #Extract only chr, startWindow, endWindow and ID# 14 not IHS\n",
    "\n",
    "\n",
    "# FST\n",
    "\n",
    "dfTemporal <- NULL\n",
    "pops <- c(\"CEU2CHB\", \"CEU2YRI\", \"YRI2CHB\")\n",
    "chr <- c(1:22, \"X\", \"Y\")\n",
    "for (i in pops) {\n",
    "    if (i == \"CEU2CHB\") {\n",
    "        windowsToStat <- windows100kb[windows100kb[[\"pop\"]] == \"CEU\" | windows100kb[[\"pop\"]] == \n",
    "            \"CHB\", c(\"chr\", \"startW\", \"endW\", \"ID\", \"pop\")]\n",
    "        windowsToStat <- unique(windowsToStat[duplicated(windowsToStat[, \"ID\"]), \n",
    "            c(\"chr\", \"startW\", \"endW\", \"ID\")])\n",
    "        groupA <- windowsToStat\n",
    "        groupA <- unique(groupA)\n",
    "        setkey(groupA, chr, startW, endW)\n",
    "        groupA <- na.omit(groupA)\n",
    "    } else if (i == \"CEU2YRI\") {\n",
    "        windowsToStat <- windows100kb[windows100kb[[\"pop\"]] == \"CEU\" | windows100kb[[\"pop\"]] == \n",
    "            \"YRI\", c(\"chr\", \"startW\", \"endW\", \"ID\", \"pop\")]\n",
    "        windowsToStat <- unique(windowsToStat[duplicated(windowsToStat[, \"ID\"]), \n",
    "            c(\"chr\", \"startW\", \"endW\", \"ID\")])\n",
    "        groupA <- as.data.table(windowsToStat)\n",
    "        groupA <- unique(groupA)\n",
    "        setkey(groupA, chr, startW, endW)\n",
    "        groupA <- na.omit(groupA)\n",
    "    } else if (i == \"YRI2CHB\") {\n",
    "        windowsToStat <- windows100kb[windows100kb[[\"pop\"]] == \"YRI\" | windows100kb[[\"pop\"]] == \n",
    "            \"CHB\", c(\"chr\", \"startW\", \"endW\", \"ID\", \"pop\")]\n",
    "        windowsToStat <- unique(windowsToStat[duplicated(windowsToStat[, \"ID\"]), \n",
    "            c(\"chr\", \"startW\", \"endW\", \"ID\")])\n",
    "        groupA <- as.data.table(windowsToStat)\n",
    "        groupA <- unique(groupA)\n",
    "        setkey(groupA, chr, startW, endW)\n",
    "        groupA <- na.omit(groupA)\n",
    "    }\n",
    "    print(\"+++++++++++++++++\")\n",
    "    print(i)\n",
    "    print(\"+++++++++++++++++\")\n",
    "    for (j in chr) {\n",
    "        print(j)\n",
    "        folder <- paste0(\"~/pophuman.uab.cat/Data/Fst/\", j, \"/\", i, \"chr\", j, \".weir.fst\")\n",
    "        df <- fread(folder, header = T, sep = \"\\t\")\n",
    "        colnames(df) <- c(\"chr\", \"start\", \"Fst\")\n",
    "        df[[\"start\"]] <- df[[\"start\"]]\n",
    "        df[[\"end\"]] <- df[[\"start\"]]\n",
    "        df[[\"chr\"]] <- paste0(\"chr\", j)\n",
    "        df[[\"pop\"]] <- i\n",
    "        \n",
    "        groupB <- df\n",
    "        setkey(groupB, chr, start, end)\n",
    "        \n",
    "        over <- foverlaps(groupA, groupB, nomatch = 0)\n",
    "        over <- unique(over)\n",
    "        \n",
    "        dfTemporal <- rbind(dfTemporal, over)\n",
    "        \n",
    "    }\n",
    "}\n",
    "\n",
    "fst100kb <- dfTemporal %>% group_by(ID, pop, startW, endW, chr) %>% summarise(Fst = mean(na.omit(Fst)))\n",
    "fst100kb <- as.data.table(fst100kb)\n",
    "\n",
    "windowsID <- unique(windows100kb[, c(\"chr\", \"startW\", \"endW\", \"ID\")])\n",
    "windows100kb <- merge(as.data.frame(fst100kb), as.data.frame(windows100kb), by = c(\"ID\", \n",
    "    \"pop\", \"chr\", \"startW\", \"endW\", \"Fst\"), all = T)\n",
    "\n",
    "windows100kb <- as.data.table(windows100kb)\n",
    "# Overwrite existing data.frames\n",
    "save(windows100kb, file = \"~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData\")\n",
    "fwrite(windows100kb, file = \"~/pophuman.uab.cat/Results/100kb/tabFiles/windowsStatisticsPopulations100kb.tab\", \n",
    "    quote = F, col.names = T, row.names = F, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-05-23T14:31:14.392Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "head(windows100kb)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Associate genes to windows and IDs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Opening windows, statisitics and populations data.frame and merge with a processed [Gencode release 28](https://www.gencodegenes.org/releases/28lift37.html) gene annotations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T10:32:09.493389Z",
     "start_time": "2018-04-16T10:34:35.652Z"
    },
    "code_folding": [
     1
    ],
    "hidden": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "genes <- fread(\"~/pophuman.uab.cat/Data/genesCoordinates.txt\", header = T)  #NEED TO UPLOAD THE FILE ANYWHERE\n",
    "if (!exists(\"windows\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/windowsStatisticsPopulations.RData\")\n",
    "}\n",
    "\n",
    "groupA <- as.data.table(windows[, c(\"ID\", \"chr\", \"startW\", \"endW\")])\n",
    "groupA <- unique(groupA)\n",
    "setkey(groupA, chr, startW, endW)\n",
    "groupB <- genes[, c(\"GeneID\", \"chr\", \"start1000\", \"end1000\")]\n",
    "setkey(groupB, chr, start1000, end1000)\n",
    "\n",
    "over <- foverlaps(groupA, groupB, nomatch = 0)\n",
    "genes2windows <- over\n",
    "\n",
    "save(genes2windows, file = \"~/pophuman.uab.cat/Results/10kb/genes2windows.RData\")\n",
    "fwrite(over, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/genes2windows.tab\", quote = F, \n",
    "    col.names = T, row.names = F, sep = \"\\t\")\n",
    "\n",
    "genes2windowsID <- unique(genes2windows[, c(\"GeneID\", \"ID\")])\n",
    "genesSymbol <- fread(\"~/pophuman.uab.cat/Data/genes_ensembl_symbol\", header = T)  #NEED TO UPLOAD THE FILE ANYWHERE\n",
    "genesSymbol <- unique(genesSymbol)\n",
    "colnames(genesSymbol) <- c(\"GeneID\", \"symbol\")\n",
    "\n",
    "genes2windowsSymbol <- merge(genes2windowsID, genesSymbol, by = \"GeneID\")\n",
    "genes2windowsSymbol <- genes2windowsSymbol[, c(\"ID\", \"symbol\")]\n",
    "genes2windowsSymbol <- unique(genes2windowsSymbol)\n",
    "\n",
    "# Summarise GeneID by ID with dplyr\n",
    "symbolPerWindow <- genes2windowsSymbol %>% group_by(ID) %>% summarise(GeneID = paste(symbol, \n",
    "    collapse = \",\"))\n",
    "symbolPerWindow <- as.data.table(symbolPerWindow)\n",
    "\n",
    "save(symbolPerWindow, file = \"~/pophuman.uab.cat/Results/10kb/symbolPerWindow.RData\")\n",
    "fwrite(symbolPerWindow, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/symbolPerWindow.tab\", \n",
    "    quote = F, col.names = T, row.names = F, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Associate tads in windows and statistics value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T10:35:42.919593Z",
     "start_time": "2018-04-16T10:35:42.898Z"
    },
    "hidden": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "if (!exists(\"windows\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/windowsStatisticsPopulations.RData\")\n",
    "}\n",
    "subsetWindows <- unique(as.data.table(windows[, c(\"ID\", \"chr\", \"startW\", \"endW\")]))\n",
    "\n",
    "load(\"~/pophuman.uab.cat/Results/10kb/symbolPerWindow.RData\")\n",
    "genes2windows <- as.data.table(symbolPerWindow)\n",
    "\n",
    "tads <- fread(\"~/pophuman.uab.cat/Data/TADs/GM12878_Lieberman-raw_TADs.txt\", header = F)  #NEED TO DOWNLOAD\n",
    "colnames(tads) <- c(\"chr\", \"startTAD\", \"endTAD\")\n",
    "tads[, `:=`(IDTAD, .GRP), by = .(chr, startTAD, endTAD)]\n",
    "\n",
    "\n",
    "groupA <- unique(tads)\n",
    "setkey(groupA, chr, startTAD, endTAD)\n",
    "groupB <- subsetWindows\n",
    "groupB <- unique(groupB)\n",
    "setkey(groupB, chr, startW, endW)\n",
    "\n",
    "overTads <- foverlaps(groupA, groupB, nomatch = 0)\n",
    "\n",
    "tads2windows <- merge(overTads, genes2windows, by = c(\"ID\"), all = T)\n",
    "tads2windows <- merge(tads2windows, windows, by = c(\"ID\", \"chr\", \"startW\", \"endW\"), \n",
    "    all = T)\n",
    "\n",
    "save(tads2windows, file = \"~/pophuman.uab.cat/Results/10kb/tads2windows.RData\")\n",
    "write.table(tads2windows, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/tads2windows.tab\", \n",
    "    quote = F, col.names = T, row.names = F, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Associate genes to windows, statisitics and populations data.frame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-29T14:55:00.998820Z",
     "start_time": "2018-05-29T14:54:31.636Z"
    },
    "code_folding": [],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if (!exists(\"windowsStatisticsPopulations\") | !exists(\"symbolPerWindow\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/windowsStatisticsPopulations.RData\")\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/symbolPerWindow.RData\")\n",
    "}\n",
    "\n",
    "# Cleaning table\n",
    "genes2tad2windows <- windows\n",
    "fst.xpehh <- subset(genes2tad2windows, pop == \"CEU2CHB\" | pop == \"CEU2YRI\" | pop == \n",
    "    \"YRI2CHB\")\n",
    "dim(fst.xpehh)\n",
    "fst.xpehh <- merge(as.data.frame(fst.xpehh), as.data.frame(symbolPerWindow), by = \"ID\", \n",
    "    all = T)\n",
    "dim(fst.xpehh)\n",
    "dim(windows)\n",
    "genes2tad2windows <- merge(as.data.frame(genes2tad2windows), as.data.frame(symbolPerWindow), \n",
    "    by = \"ID\", all = T)\n",
    "dim(genes2tad2windows)\n",
    "genes2tad2windows <- as.data.table(genes2tad2windows)\n",
    "genes2tad2windowsCleaned <- genes2tad2windows[as.numeric(genes2tad2windows[[\"S\"]]) >= \n",
    "    5 & !is.na(genes2tad2windows[[\"S\"]]), ]\n",
    "genes2tad2windows <- merge(as.data.frame(genes2tad2windowsCleaned), as.data.frame(fst.xpehh), \n",
    "    all = T)\n",
    "genes2tad2windows <- as.data.table(genes2tad2windows)\n",
    "\n",
    "# Be sure each field has the proper type\n",
    "genes2tad2windows[[\"pop\"]] <- as.character(genes2tad2windows[[\"pop\"]])\n",
    "genes2tad2windows[[\"GeneID\"]] <- as.character(genes2tad2windows[[\"GeneID\"]])\n",
    "\n",
    "save(genes2tad2windows, file = \"~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData\")\n",
    "fwrite(genes2tad2windows, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/genes2tad2windows.tab\", \n",
    "    quote = F, col.names = T, row.names = F, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### Extracting needed features"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "**Extract basic features in independents data.frames in order to use in statistics calculations or plots**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "#### Coordinates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-30T08:03:14.487317Z",
     "start_time": "2018-05-30T08:00:39.765Z"
    },
    "hidden": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "if (!exists(\"genes2tad2windows\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData\")\n",
    "}\n",
    "\n",
    "windowsCoord <- unique(genes2tad2windows[, c(\"ID\", \"chr\", \"startW\", \"endW\")])\n",
    "save(windowsCoord, file = \"~/pophuman.uab.cat/Results/10kb/windowsCoord.RData\")\n",
    "fwrite(windowsCoord, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/windowsCoord.tab\", \n",
    "    quote = F, col.names = T, row.names = F, sep = \"\\t\")\n",
    "\n",
    "coordinates <- unique(genes2tad2windows[, c(\"chr\", \"ID\", \"startW\", \"endW\")])\n",
    "save(coordinates, file = \"~/pophuman.uab.cat/Results/10kb/coordinates.RData\")\n",
    "fwrite(coordinates, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/coordinates.tab\", \n",
    "    quote = F, col.names = T, row.names = F, sep = \"\\t\")\n",
    "\n",
    "\n",
    "genes2ID <- unique(genes2tad2windows[, c(\"ID\", \"GeneID\")])\n",
    "\n",
    "number <- 1\n",
    "for (i in 1:nrow(genes2ID)) {\n",
    "    \n",
    "    if (is.na(genes2ID[[\"GeneID\"]][i]) | genes2ID[[\"GeneID\"]][i] == \"NA\") {\n",
    "        print(i)\n",
    "        genes2ID[[\"GeneID\"]][i] <- paste0(\"nonCoding\", number)\n",
    "        number <- number + 1\n",
    "        print(genes2ID[[\"GeneID\"]][i])\n",
    "    }\n",
    "}\n",
    "genes2ID\n",
    "save(genes2ID, file = \"~/pophuman.uab.cat/Results/10kb/genes2ID.RData\")\n",
    "fwrite(genes2ID, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/genes2ID.tab\", quote = F, \n",
    "    col.names = T, row.names = F, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "#### Recombination by windows"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "<p style='text-align: justify;'>Calculating recombination in **cM/MB** by **unique windows** using recombination data from [Bhèrer et al.,(2017)](https://www.nature.com/articles/ncomms14994) and using their own procedure. Raw data and scripts were retrieved from [cbherer GitHub](https://github.com/cbherer/Bherer_etal_SexualDimorphismRecombination).</p>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "##### 10-kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-27T16:27:00.625264Z",
     "start_time": "2018-03-27T13:25:49.914Z"
    },
    "code_folding": [
     0,
     6,
     22,
     32,
     53
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "getRate <- function(map, intI) {\n",
    "    # get recombination rate in cM/Mb for a given map and interval\n",
    "    rr1 <- approx(map$pos, map$cM, xout = intI[, 1])\n",
    "    rr2 <- approx(map$pos, map$cM, xout = intI[, 2])\n",
    "    rate <- (rr2$y - rr1$y)/((rr2$x - rr1$x)/1e+06)  # cM/Mb\n",
    "    return(rate)\n",
    "}\n",
    "binGenome_per_chr <- function(chr, bsize, rm.gaps = TRUE) {\n",
    "    # zero-based indexing, half open intervals\n",
    "    bpr <- c(1, genomeData$nbases[genomeData$chrom == chr])\n",
    "    bpr[1] <- bpr[1] - bpr[1]%%bsize\n",
    "    bpr[2] <- bpr[2] + (bsize - bpr[2]%%bsize)\n",
    "    x <- seq(bpr[1], bpr[2], by = bsize)\n",
    "    seqbinI <- Intervals(cbind(x[-length(x)], x[-1]), closed = c(TRUE, FALSE), type = \"Z\")\n",
    "    if (rm.gaps) {\n",
    "        gapI <- with(gap[gap$chrom == chr, ], {\n",
    "            Intervals(cbind(chromStart, chromEnd), closed = c(TRUE, FALSE), type = \"Z\")\n",
    "        })\n",
    "        ol <- unlist(interval_overlap(from = gapI, to = seqbinI))\n",
    "        if (length(ol) > 0) \n",
    "            seqbinI <- seqbinI[-ol, ]\n",
    "    }\n",
    "    return(seqbinI)\n",
    "}\n",
    "\n",
    "if (!exists(\"genes2tad2windows\") | !exists(\"windowsCoord\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData\")\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/windowsCoord.RData\")\n",
    "    \n",
    "}\n",
    "\n",
    "mapname <- read.table(\"~/pophuman.uab.cat/Data/Refined_genetic_map_b37/sexavg_chr1.txt\", \n",
    "    header = T)  # Refined_genetic_map_b37/sexavg_chr22.txt\n",
    "chr <- c(\"1\", \"2\", \"3\", \"4\", \"5\", \"6\", \"7\", \"8\", \"9\", \"10\", \"11\", \"12\", \"13\", \"14\", \n",
    "    \"15\", \"16\", \"17\", \"18\", \"19\", \"20\", \"21\", \"22\")\n",
    "\n",
    "df.recomb <- NULL\n",
    "for (i in chr) {\n",
    "    mapname <- paste0(\"~/pophuman.uab.cat/Data/Refined_genetic_map_b37/sexavg_chr\", \n",
    "        i, \".txt\")\n",
    "    map <- fread(mapname, header = T)\n",
    "    print(paste0(\"chr\", i))\n",
    "    windowsChr <- windowsCoord[windowsCoord[[\"chr\"]] == paste0(\"chr\", i), ]\n",
    "    for (j in 1:length(windowsChr[[\"startW\"]])) {\n",
    "        # print(j)\n",
    "        start <- as.numeric(windowsChr[j, 3])\n",
    "        end <- as.numeric(windowsChr[j, 4])\n",
    "        chr <- as.character(windowsChr[j, 2])\n",
    "        ID <- as.character(windowsChr[j, 1])\n",
    "        \n",
    "        intI = data.frame(start, end)\n",
    "        \n",
    "        recomb <- getRate(map, intI)\n",
    "        tmp <- c(ID, chr, start, end, recomb)\n",
    "        \n",
    "        df.recomb <- rbind(df.recomb, tmp)\n",
    "    }\n",
    "}\n",
    "\n",
    "chr <- c(\"X\")\n",
    "for (i in chr) {\n",
    "    mapname <- paste0(\"~/pophuman.uab.cat/Data/Refined_genetic_map_b37/female_chr\", \n",
    "        i, \".txt\")\n",
    "    map <- fread(mapname, header = T)\n",
    "    print(paste0(\"chr\", i))\n",
    "    windowsChr <- windowsCoord[windowsCoord[[\"chr\"]] == paste0(\"chr\", i), ]\n",
    "    for (j in 1:length(windowsChr[[\"startW\"]])) {\n",
    "        print(j)\n",
    "        start <- as.numeric(windowsChr[j, 3])\n",
    "        end <- as.numeric(windowsChr[j, 4])\n",
    "        ID <- as.character(windowsChr[j, 2])\n",
    "        chr <- as.character(windowsChr[j, 1])\n",
    "        \n",
    "        intI = data.frame(start, end)\n",
    "        \n",
    "        recomb <- getRate(map, intI)\n",
    "        tmp <- c(ID, chr, start, end, recomb)\n",
    "        \n",
    "        df.recomb <- rbind(df.recomb, tmp)\n",
    "    }\n",
    "}\n",
    "\n",
    "df.recomb <- as.data.table(df.recomb, row.names = F)\n",
    "\n",
    "colnames(df.recomb) <- c(\"ID\", \"chr\", \"startW\", \"endW\", \"recomb\")\n",
    "windowsRecomb <- df.recomb\n",
    "\n",
    "windowsRecomb[[\"ID\"]] <- as.integer(windowsRecomb[[\"ID\"]])\n",
    "windowsRecomb[[\"startW\"]] <- as.integer(windowsRecomb[[\"startW\"]])\n",
    "windowsRecomb[[\"endW\"]] <- as.integer(windowsRecomb[[\"endW\"]])\n",
    "windowsRecomb[[\"recomb\"]] <- as.numeric(windowsRecomb[[\"recomb\"]])\n",
    "\n",
    "save(windowsRecomb, file = \"~/pophuman.uab.cat/Results/10kb/windowsRecomb.RData\")\n",
    "fwrite(windowsRecomb, \"~/pophuman.uab.cat/Results/10kb/tabFiles/windowsRecomb.tab\", \n",
    "    row.names = F, col.names = T, sep = \"\\t\", quote = F)\n",
    "\n",
    "# Adding recomb to genes2tad2windows\n",
    "genes2tad2windows <- merge(genes2tad2windows, windowsRecomb, by = c(\"ID\", \"startW\", \n",
    "    \"endW\", \"chr\"), all.x = T)\n",
    "save(genes2tad2windows, file = \"~/pophuman.uab.cat/Results/10kb/genes2tads2windows2.RData\")\n",
    "fwrite(genes2tad2windows, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/genes2tads2windows2.tab\", \n",
    "    row.names = F, col.names = T, sep = \"\\t\", quote = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "##### 100-kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "getRate <- function(map, intI) {\n",
    "    # get recombination rate in cM/Mb for a given map and interval\n",
    "    rr1 <- approx(map$pos, map$cM, xout = intI[, 1])\n",
    "    rr2 <- approx(map$pos, map$cM, xout = intI[, 2])\n",
    "    rate <- (rr2$y - rr1$y)/((rr2$x - rr1$x)/1e+06)  # cM/Mb\n",
    "    return(rate)\n",
    "}\n",
    "binGenome_per_chr <- function(chr, bsize, rm.gaps = TRUE) {\n",
    "    # zero-based indexing, half open intervals\n",
    "    bpr <- c(1, genomeData$nbases[genomeData$chrom == chr])\n",
    "    bpr[1] <- bpr[1] - bpr[1]%%bsize\n",
    "    bpr[2] <- bpr[2] + (bsize - bpr[2]%%bsize)\n",
    "    x <- seq(bpr[1], bpr[2], by = bsize)\n",
    "    seqbinI <- Intervals(cbind(x[-length(x)], x[-1]), closed = c(TRUE, FALSE), type = \"Z\")\n",
    "    if (rm.gaps) {\n",
    "        gapI <- with(gap[gap$chrom == chr, ], {\n",
    "            Intervals(cbind(chromStart, chromEnd), closed = c(TRUE, FALSE), type = \"Z\")\n",
    "        })\n",
    "        ol <- unlist(interval_overlap(from = gapI, to = seqbinI))\n",
    "        if (length(ol) > 0) \n",
    "            seqbinI <- seqbinI[-ol, ]\n",
    "    }\n",
    "    return(seqbinI)\n",
    "}\n",
    "\n",
    "if (!exists(\"windows100kb\") | !exists(\"windowsCoord\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData\")\n",
    "    load(\"~/pophuman.uab.cat/Results/100kb/windowsCoord100kb.RData\")\n",
    "    \n",
    "}\n",
    "\n",
    "mapname <- read.table(\"~/pophuman.uab.cat/Data/Refined_genetic_map_b37/sexavg_chr1.txt\", \n",
    "    header = T)  # Refined_genetic_map_b37/sexavg_chr22.txt\n",
    "chr <- c(\"1\", \"2\", \"3\", \"4\", \"5\", \"6\", \"7\", \"8\", \"9\", \"10\", \"11\", \"12\", \"13\", \"14\", \n",
    "    \"15\", \"16\", \"17\", \"18\", \"19\", \"20\", \"21\", \"22\")\n",
    "\n",
    "df.recomb <- NULL\n",
    "for (i in chr) {\n",
    "    mapname <- paste0(\"~/pophuman.uab.cat/Data/Refined_genetic_map_b37/sexavg_chr\", \n",
    "        i, \".txt\")\n",
    "    map <- fread(mapname, header = T)\n",
    "    print(paste0(\"chr\", i))\n",
    "    windowsChr <- windowsCoord100kb[windowsCoord100kb[[\"chr\"]] == paste0(\"chr\", i), \n",
    "        ]\n",
    "    for (j in 1:length(windowsChr[[\"startW\"]])) {\n",
    "        # print(j)\n",
    "        start <- as.numeric(windowsChr[j, 3])\n",
    "        end <- as.numeric(windowsChr[j, 4])\n",
    "        chr <- as.character(windowsChr[j, 2])\n",
    "        ID <- as.character(windowsChr[j, 1])\n",
    "        \n",
    "        intI = data.frame(start, end)\n",
    "        \n",
    "        recomb <- getRate(map, intI)\n",
    "        tmp <- c(ID, chr, start, end, recomb)\n",
    "        \n",
    "        df.recomb <- rbind(df.recomb, tmp)\n",
    "    }\n",
    "}\n",
    "\n",
    "chr <- c(\"X\")\n",
    "for (i in chr) {\n",
    "    mapname <- paste0(\"~/pophuman.uab.cat/Data/Refined_genetic_map_b37/female_chr\", \n",
    "        i, \".txt\")\n",
    "    map <- fread(mapname, header = T)\n",
    "    print(paste0(\"chr\", i))\n",
    "    windowsChr <- windowsCoord100kb[windowsCoord100kb[[\"chr\"]] == paste0(\"chr\", i), \n",
    "        ]\n",
    "    for (j in 1:length(windowsChr[[\"startW\"]])) {\n",
    "        print(j)\n",
    "        start <- as.numeric(windowsChr[j, 3])\n",
    "        end <- as.numeric(windowsChr[j, 4])\n",
    "        ID <- as.character(windowsChr[j, 2])\n",
    "        chr <- as.character(windowsChr[j, 1])\n",
    "        \n",
    "        intI = data.frame(start, end)\n",
    "        \n",
    "        recomb <- getRate(map, intI)\n",
    "        tmp <- c(ID, chr, start, end, recomb)\n",
    "        \n",
    "        df.recomb <- rbind(df.recomb, tmp)\n",
    "    }\n",
    "}\n",
    "\n",
    "df.recomb <- as.data.table(df.recomb, row.names = F)\n",
    "\n",
    "colnames(df.recomb) <- c(\"ID\", \"chr\", \"startW\", \"endW\", \"recomb\")\n",
    "windowsRecomb <- df.recomb\n",
    "\n",
    "windowsRecomb[[\"ID\"]] <- as.integer(windowsRecomb[[\"ID\"]])\n",
    "windowsRecomb[[\"startW\"]] <- as.integer(windowsRecomb[[\"startW\"]])\n",
    "windowsRecomb[[\"endW\"]] <- as.integer(windowsRecomb[[\"endW\"]])\n",
    "windowsRecomb[[\"recomb\"]] <- as.numeric(windowsRecomb[[\"recomb\"]])\n",
    "\n",
    "save(windowsRecomb, file = \"~/pophuman.uab.cat/Results/100kb/windowsRecomb100kb.RData\")\n",
    "fwrite(windowsRecomb, \"~/pophuman.uab.cat/Results/100kb/tabFiles/windowsRecomb100kb.tab\", \n",
    "    row.names = F, col.names = T, sep = \"\\t\", quote = F)\n",
    "\n",
    "# Adding recomb to genes2tad2windows\n",
    "windows100kb <- merge(windows100kb, windowsRecomb, by = c(\"ID\", \"startW\", \"endW\", \n",
    "    \"chr\"), all.x = T)\n",
    "save(windows100kb, file = \"~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData\")\n",
    "fwrite(windows100kb, file = \"~/pophuman.uab.cat/Results/100kb/tabFiles/windowsStatisticsPopulations100kb.tab\", \n",
    "    row.names = F, col.names = T, sep = \"\\t\", quote = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "#### Grouping windows"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Group IDs if coordinates are contigous or distance are minor than 10-kb (single windows size) in order to maintain resolution. Distances minor than 10-kb windows correspond to position not calculated due to low quality. Will be used in plots to aggreate significatives windows in larger coordinates regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if (!exists(\"genes2tad2windows\") && !exists(\"windowsCoord\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData\")\n",
    "}\n",
    "\n",
    "groupingWindows <- function(df) {\n",
    "    \n",
    "    # windows <- unique(df[,c(1,4,6,7)])\n",
    "    windows <- unique(df[, c(\"chr\", \"ID\", \"startW\", \"endW\")])\n",
    "    \n",
    "    # windows <- windows[ order(windows[,2], windows[,3]), ]\n",
    "    row.names(windows) <- NULL\n",
    "    \n",
    "    jump <- 1\n",
    "    for (i in 1:length(windows[[\"ID\"]])) {\n",
    "        distanceWindows <- windows[[\"startW\"]][i] - windows[[\"endW\"]][i - 1]\n",
    "        if (i == 1) {\n",
    "            jump <- jump\n",
    "            windows[[\"vector\"]][i] <- jump\n",
    "            windows[[\"distance\"]][i] <- 0\n",
    "        } else if (distanceWindows <= 10000) {\n",
    "            \n",
    "            # print(i) print(windows$endW[i]) print(windows$startW[i-1])\n",
    "            jump <- jump\n",
    "            windows[[\"vector\"]][i] <- jump\n",
    "            windows[[\"distance\"]][i] <- distanceWindows\n",
    "        } else {\n",
    "            jump <- jump + 1\n",
    "            windows[[\"vector\"]][i] <- jump\n",
    "            windows[[\"distance\"]][i] <- distanceWindows\n",
    "        }\n",
    "    }\n",
    "    names(windows)[names(windows) == \"vector\"] <- \"Group\"\n",
    "    # windows[['Group']] <- paste0('Group_', cumsum(c(1, diff(windows[['vector']]) !=\n",
    "    # 1)))\n",
    "    windows <- windows[, c(\"chr\", \"startW\", \"endW\", \"ID\", \"Group\", \"distance\")]\n",
    "    return(windows)\n",
    "}\n",
    "\n",
    "contigousWindows <- groupingWindows(genes2tad2windows)\n",
    "save(contigousWindows, file = \"~/pophuman.uab.cat/Results/10kb/contigousWindows.RData\")\n",
    "fwrite(contigousWindows, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/contigousWindows.RData\", \n",
    "    row.names = F, col.names = T, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "##### Grouping windows features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if (!exists(\"windowsCoord\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/windowsCoord.RData\")\n",
    "}\n",
    "if (!exists(\"windowsCoord\")) {\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/contigousWindows.RData\")\n",
    "}\n",
    "\n",
    "# GROUP_BY(GROUP) AND SUMMARIZE START, END AND DISTANCE TO CREATE LABELCOORD\n",
    "# USING THE MAXIMUM RANGE IN GROUP\n",
    "contigousCoord <- contigousWindows %>% group_by(Group) %>% summarize(chr = toString(unique(chr)), \n",
    "    labelCoord = paste0(chr, \":\", min(startW), \"-\", max(endW))) %>% as.data.table()\n",
    "contigousCoord\n",
    "\n",
    "contigousCoord <- merge(contigousCoord, contigousWindows, by = c(\"Group\", \"chr\"))\n",
    "\n",
    "contigousCoord <- contigousCoord[order(contigousCoord[[\"chr\"]], contigousCoord[[\"startW\"]]), \n",
    "    c(\"chr\", \"startW\", \"endW\", \"ID\", \"Group\", \"labelCoord\", \"distance\")]\n",
    "save(contigousCoord, file = \"~/pophuman.uab.cat/Results/10kb/contigousCoord.RData\")\n",
    "fwrite(contigousCoord, file = \"~/pophuman.uab.cat/Results/10kb/tabFiles/contigousCoord.RData\", \n",
    "    row.names = F, col.names = T, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Empirical p-values"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-23T10:51:39.786058Z",
     "start_time": "2018-05-23T10:54:18.031Z"
    },
    "hidden": true
   },
   "source": [
    "<p style='text-align: justify;'>In order to identify the strongest genome-wide signals of selection, we did not considered 10-kb windows containing less than 5 segregating sites. Removing windows with a low number of segregating sites allows to discard putative extreme values due to statistical noise, as well as performing a more conservative analysis. Empirical p-values were calculated as the sum of the values more extreme than a given value divided by the total number of analyzable windows for each population and statistic.</p>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### Empirical p-values calculations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-27T11:22:53.858009Z",
     "start_time": "2018-06-27T11:26:27.454Z"
    },
    "hidden": true
   },
   "source": [
    "Each function perform the empirical distribution of determinated test, checking one or two distribution tails depending on the test. Empirical p-values were calculated independently from autosome and chromosome X taking into account only the correspondent values. Chromosome Y were exclude from the analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [
     0,
     12,
     18,
     24,
     30,
     35,
     41,
     48,
     54,
     60,
     66,
     72,
     193,
     355,
     360,
     365,
     370,
     375,
     380,
     385,
     386,
     391,
     396,
     401,
     406,
     411,
     417,
     422,
     427,
     432,
     437,
     443,
     448,
     453,
     458,
     463,
     464,
     469,
     474,
     475,
     479,
     518,
     526,
     532,
     533,
     538,
     544,
     550,
     556,
     562,
     569,
     575,
     581,
     587,
     593,
     594,
     600,
     606,
     612,
     618,
     624,
     631,
     637,
     643,
     649,
     655,
     662,
     668,
     674,
     711,
     716,
     721,
     726,
     731,
     736,
     741,
     747,
     752,
     757,
     762,
     763,
     767,
     773,
     778,
     779,
     783,
     788,
     793,
     794,
     799,
     804,
     808,
     813,
     818,
     824,
     829,
     834,
     1032
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "pvalueEmpiricV1 <- function(stat, pop, autosomes, sexualX, sexualY) {\n",
    "    print(\"++++++++++++++++++++++\")\n",
    "    print(pop)\n",
    "    statPvalueFinal <- NULL\n",
    "    print(\"++++++++++++++++++++++\")\n",
    "    \n",
    "    if (autosomes == TRUE) {\n",
    "        if (pop == \"YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"ACB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ACB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"LWK\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"LWK\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ESN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ESN\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GWD\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GWD\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"MSL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"MSL\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ASW\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ASW\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GBR\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GBR\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"FIN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"FIN\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"IBS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"IBS\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"TSI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"TSI\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CDX\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CDX\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHS\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"JPT\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"JPT\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"KHV\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"KHV\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"BEB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"BEB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GIH\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GIH\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            #### REPETIR GIH!!!1 en tajima, fuli d fuli f\n",
    "        }\n",
    "        if (pop == \"ITU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ITU\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"PJL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"PJL\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"STU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"STU\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CEU2YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"YRI2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        medianStat <- median(na.omit(windows[[stat]]))\n",
    "        \n",
    "        statPvalue <- NULL\n",
    "        \n",
    "        for (window in 1:length(windows[[stat]])) {\n",
    "            print(window)\n",
    "            \n",
    "            if (is.na(windows[[stat]][window]) == T) {\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  \"NA\")\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "            } else {\n",
    "                # quantilePosition <- ecdf_fun(windows[[stat]],windows[[stat]][window])\n",
    "                if (windows[[stat]][window] > medianStat) {\n",
    "                  # if (quantilePosition > 0.5) {\n",
    "                  pvalue = sum(na.omit(windows[[stat]]) >= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                } else {\n",
    "                  pvalue = sum(na.omit(windows[[stat]]) <= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                }\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  pvalue)\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "                \n",
    "            }\n",
    "            print(result)\n",
    "        }\n",
    "        \n",
    "        statPvalue <- as.data.frame(statPvalue)\n",
    "        colnames(statPvalue) <- c(\"pop\", \"ID\", stat, \"Pvalue\")\n",
    "        #statPvalue[[\"adjust\"]] <- p.adjust(as.numeric(as.character(statPvalue[[\"Pvalue\"]])),\"fdr\")\n",
    "        row.names(statPvalue) <- NULL\n",
    "        statPvalueFinal <- rbind(statPvalueFinal,statPvalue)\n",
    "    } \n",
    "    if (sexualX == TRUE) {\n",
    "        if (pop == \"YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"ACB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ACB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"LWK\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"LWK\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ESN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ESN\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GWD\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GWD\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"MSL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"MSL\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ASW\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ASW\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GBR\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GBR\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"FIN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"FIN\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"IBS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"IBS\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"TSI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"TSI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CDX\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CDX\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHS\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"JPT\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"JPT\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"KHV\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"KHV\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"BEB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"BEB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GIH\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GIH\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            #### REPETIR GIH!!!1 en tajima, fuli d fuli f\n",
    "        }\n",
    "        if (pop == \"ITU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ITU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"PJL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"PJL\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"STU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"STU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CEU2YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"YRI2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        medianStat <- median(na.omit(windows[[stat]]))\n",
    "        \n",
    "        statPvalue <- NULL\n",
    "        \n",
    "        for (window in 1:length(windows[[stat]])) {\n",
    "            print(window)\n",
    "            \n",
    "            if (is.na(windows[[stat]][window]) == T) {\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  \"NA\")\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "            } else {\n",
    "                # quantilePosition <- ecdf_fun(windows[[stat]],windows[[stat]][window])\n",
    "                if (windows[[stat]][window] > medianStat) {\n",
    "                  # if (quantilePosition > 0.5) {\n",
    "                  pvalue = sum(na.omit(windows[[stat]]) >= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                } else {\n",
    "                  pvalue = sum(na.omit(windows[[stat]]) <= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                }\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  pvalue)\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "                \n",
    "            }\n",
    "            print(result)\n",
    "        }\n",
    "        statPvalue <- as.data.frame(statPvalue)\n",
    "        colnames(statPvalue) <- c(\"pop\", \"ID\", stat, \"Pvalue\")\n",
    "        #statPvalue[[\"adjust\"]] <- p.adjust(as.numeric(as.character(statPvalue[[\"Pvalue\"]])),\"fdr\")\n",
    "        row.names(statPvalue) <- NULL\n",
    "        statPvalueFinal <- rbind(statPvalueFinal,statPvalue)\n",
    "    }\n",
    "    if (sexualY == TRUE) {\n",
    "        if (pop == \"YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "        }\n",
    "        if (pop == \"ACB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ACB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"LWK\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"LWK\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ESN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ESN\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GWD\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GWD\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"MSL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"MSL\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ASW\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ASW\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GBR\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GBR\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"FIN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"FIN\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"IBS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"IBS\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"TSI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"TSI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CDX\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CDX\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHS\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"JPT\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"JPT\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"KHV\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"KHV\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"BEB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"BEB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GIH\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GIH\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            #### REPETIR GIH!!!1 en tajima, fuli d fuli f\n",
    "        }\n",
    "        if (pop == \"ITU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ITU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"PJL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"PJL\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"STU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"STU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CEU2YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"YRI2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        medianStat <- median(na.omit(windows[[stat]]))\n",
    "        \n",
    "        statPvalue <- NULL\n",
    "        \n",
    "        for (window in 1:length(windows[[stat]])) {\n",
    "            print(window)\n",
    "            \n",
    "            if (is.na(windows[[stat]][window]) == T) {\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  \"NA\")\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "            } else {\n",
    "                # quantilePosition <- ecdf_fun(windows[[stat]],windows[[stat]][window])\n",
    "                if (windows[[stat]][window] > medianStat) {\n",
    "                  # if (quantilePosition > 0.5) {\n",
    "                  pvalue = sum(na.omit(windows[[stat]]) >= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                } else {\n",
    "                  pvalue = sum(na.omit(windows[[stat]]) <= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                }\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  pvalue)\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "                \n",
    "            }\n",
    "            print(result)\n",
    "        }\n",
    "        statPvalue <- as.data.frame(statPvalue)\n",
    "        colnames(statPvalue) <- c(\"pop\", \"ID\", stat, \"Pvalue\")\n",
    "        #statPvalue[[\"adjust\"]] <- p.adjust(as.numeric(as.character(statPvalue[[\"Pvalue\"]])),\"fdr\")\n",
    "        row.names(statPvalue) <- NULL\n",
    "    }\n",
    "    return(statPvalueFinal)\n",
    "}  # for tajima's, fay, fu lis\n",
    "pvalueEmpiricV2 <- function(stat, pop, autosomes, sexual) {\n",
    "    print(\"++++++++++++++++++++++\")\n",
    "    print(pop)\n",
    "    statPvalue <- NULL\n",
    "    statPvalueFinal <- NULL\n",
    "    print(\"++++++++++++++++++++++\")\n",
    "    \n",
    "    if (autosomes == TRUE) {\n",
    "        if (pop == \"YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ACB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ACB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"LWK\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"LWK\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ESN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ESN\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GWD\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GWD\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"MSL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"MSL\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ASW\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ASW\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GBR\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GBR\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"FIN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"FIN\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"IBS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"IBS\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"TSI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"TSI\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CDX\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CDX\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHS\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"JPT\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"JPT\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"KHV\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"KHV\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"BEB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"BEB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GIH\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GIH\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            #### REPETIR GIH!!!1 en tajima, fuli d fuli f\n",
    "        }\n",
    "        if (pop == \"ITU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ITU\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"PJL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"PJL\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"STU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"STU\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CEU2YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"YRI2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        medianStat <- median(na.omit(windows[[stat]]))\n",
    "        \n",
    "        statPvalue <- NULL\n",
    "        \n",
    "        for (window in 1:length(windows[[stat]])) {\n",
    "            print(window)\n",
    "            \n",
    "            if (is.na(windows[[stat]][window]) == T) {\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  \"NA\")\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "            } else {\n",
    "                \n",
    "                pvalue = sum(na.omit(windows[[stat]]) <= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                \n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  pvalue)\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "                \n",
    "            }\n",
    "            print(result)\n",
    "        }\n",
    "        statPvalue <- as.data.frame(statPvalue)\n",
    "        colnames(statPvalue) <- c(\"pop\", \"ID\", stat, \"Pvalue\")\n",
    "        statPvalue[[\"adjust\"]] <- p.adjust(as.numeric(as.character(statPvalue[[\"Pvalue\"]])), \n",
    "            \"fdr\")\n",
    "        row.names(statPvalue) <- NULL\n",
    "        statPvalueFinal <- rbind(statPvalueFinal,statPvalue)\n",
    "    } \n",
    "    if (sexualX == TRUE) {\n",
    "        if (pop == \"YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ACB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ACB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"LWK\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"LWK\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ESN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ESN\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GWD\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GWD\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"MSL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"MSL\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ASW\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ASW\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GBR\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GBR\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"FIN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"FIN\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"IBS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"IBS\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"TSI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"TSI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CDX\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CDX\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHS\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"JPT\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"JPT\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"KHV\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"KHV\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"BEB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"BEB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GIH\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GIH\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"ITU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ITU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"PJL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"PJL\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"STU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"STU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CEU2YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"YRI2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "            \n",
    "        }\n",
    "             \n",
    "        medianStat <- median(na.omit(windows[[stat]]))\n",
    "        \n",
    "        statPvalue <- NULL\n",
    "        \n",
    "        for (window in 1:length(windows[[stat]])) {\n",
    "            print(window)\n",
    "            \n",
    "            if (is.na(windows[[stat]][window]) == T) {\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  \"NA\")\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "            } else {\n",
    "                \n",
    "                pvalue = sum(na.omit(windows[[stat]]) <= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                \n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  pvalue)\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "                \n",
    "            }\n",
    "            print(result)\n",
    "        }\n",
    "        statPvalue <- as.data.frame(statPvalue)\n",
    "        colnames(statPvalue) <- c(\"pop\", \"ID\", stat, \"Pvalue\")\n",
    "        #statPvalue[[\"adjust\"]] <- p.adjust(as.numeric(as.character(statPvalue[[\"Pvalue\"]])), \n",
    "            \"fdr\")\n",
    "        row.names(statPvalue) <- NULL\n",
    "        statPvalueFinal <- rbind(statPvalueFinal,statPvalue)\n",
    "        \n",
    "    }\n",
    "    if (sexualY == TRUE) {\n",
    "        if (pop == \"YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ACB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ACB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"LWK\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"LWK\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ESN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ESN\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GWD\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GWD\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"MSL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"MSL\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"ASW\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ASW\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GBR\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GBR\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"FIN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"FIN\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"IBS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"IBS\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"TSI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"TSI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CDX\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CDX\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CHS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHS\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"JPT\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"JPT\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"KHV\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"KHV\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"BEB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"BEB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"GIH\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GIH\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "        }\n",
    "        if (pop == \"ITU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ITU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"PJL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"PJL\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"STU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"STU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"CEU2YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        if (pop == \"YRI2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrY\", ]\n",
    "            \n",
    "        }\n",
    "        \n",
    "        \n",
    "        medianStat <- median(na.omit(windows[[stat]]))\n",
    "        \n",
    "        statPvalue <- NULL\n",
    "        \n",
    "        for (window in 1:length(windows[[stat]])) {\n",
    "            print(window)\n",
    "            \n",
    "            if (is.na(windows[[stat]][window]) == T) {\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  \"NA\")\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "            } else {\n",
    "                \n",
    "                pvalue = sum(na.omit(windows[[stat]]) <= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                \n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  pvalue)\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "                \n",
    "            }\n",
    "            print(result)\n",
    "        }\n",
    "        statPvalue <- as.data.frame(statPvalue)\n",
    "        colnames(statPvalue) <- c(\"pop\", \"ID\", stat, \"Pvalue\")\n",
    "        #statPvalue[[\"adjust\"]] <- p.adjust(as.numeric(as.character(statPvalue[[\"Pvalue\"]])),\"fdr\")\n",
    "        row.names(statPvalue) <- NULL\n",
    "        statPvalueFinal <- rbind(statPvalueFinal,statPvalue)\n",
    "\n",
    "    }\n",
    "    return(statPvalue)\n",
    "}  # for theta and pi\n",
    "pvalueEmpiricV3 <- function(stat, pop, autosomes, sexual) {\n",
    "    print(\"++++++++++++++++++++++\")\n",
    "    print(pop)\n",
    "    statPvalue <- NULL\n",
    "    print(\"++++++++++++++++++++++\")\n",
    "    \n",
    "    if (autosomes == TRUE) {\n",
    "        if (pop == \"YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"ACB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ACB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"LWK\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"LWK\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"ESN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ESN\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"GWD\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GWD\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"MSL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"MSL\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"ASW\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ASW\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"GBR\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GBR\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"FIN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"FIN\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"IBS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"IBS\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"TSI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"TSI\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        \n",
    "        if (pop == \"CDX\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CDX\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"CHS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHS\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"JPT\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"JPT\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"KHV\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"KHV\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        \n",
    "        if (pop == \"BEB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"BEB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"GIH\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GIH\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]  #### REPETIR GIH!!!1 en tajima, fuli d fuli f\n",
    "        }\n",
    "        if (pop == \"ITU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ITU\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"PJL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"PJL\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"STU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"STU\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"CEU2YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        if (pop == \"YRI2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] != \"chrX\" & windows[[\"chr\"]] != \"chrY\", \n",
    "                ]\n",
    "        }\n",
    "        \n",
    "        \n",
    "        # medianStat <- median(na.omit(windows[[stat]]))\n",
    "        \n",
    "        \n",
    "        for (window in 1:length(windows[[stat]])) {\n",
    "            print(unique(windows[[\"chr\"]]))\n",
    "            \n",
    "            if (is.na(windows[[stat]][window]) == T) {\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  \"NA\")\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "            } else {\n",
    "                \n",
    "                pvalue = sum(na.omit(windows[[stat]]) >= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                \n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  pvalue)\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "                \n",
    "            }\n",
    "            print(result)\n",
    "        }\n",
    "        statPvalue <- as.data.frame(statPvalue)\n",
    "        colnames(statPvalue) <- c(\"pop\", \"ID\", stat, \"Pvalue\")\n",
    "        statPvalue[[\"adjust\"]] <- p.adjust(as.numeric(as.character(statPvalue[[\"Pvalue\"]])), \n",
    "            \"fdr\")\n",
    "        row.names(statPvalue) <- NULL\n",
    "    }\n",
    "    if (sexual == TRUE) {\n",
    "        if (pop == \"YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"ACB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ACB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"LWK\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"LWK\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"ESN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ESN\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"GWD\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GWD\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"MSL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"MSL\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"ASW\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ASW\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"GBR\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GBR\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"FIN\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"FIN\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"IBS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"IBS\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"TSI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"TSI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        \n",
    "        if (pop == \"CDX\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CDX\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"CHS\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CHS\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"JPT\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"JPT\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"KHV\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"KHV\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        \n",
    "        if (pop == \"BEB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"BEB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"GIH\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"GIH\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]  #### REPETIR GIH!!!1 en tajima, fuli d fuli f\n",
    "        }\n",
    "        if (pop == \"ITU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"ITU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"PJL\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"PJL\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"STU\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"STU\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        \n",
    "        if (pop == \"CEU2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"CEU2YRI\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"CEU2YRI\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        if (pop == \"YRI2CHB\") {\n",
    "            windows <- subset(genes2tad2windows, genes2tad2windows[[\"pop\"]] == \"YRI2CHB\")\n",
    "            windows <- windows[windows[[\"chr\"]] == \"chrX\", ]\n",
    "        }\n",
    "        \n",
    "        \n",
    "        # medianStat <- median(na.omit(windows[[stat]]))\n",
    "        \n",
    "        for (window in 1:length(windows[[stat]])) {\n",
    "            print(unique(windows[[\"chr\"]]))\n",
    "            \n",
    "            if (is.na(windows[[stat]][window]) == T) {\n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  \"NA\")\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "            } else {\n",
    "                \n",
    "                pvalue = sum(na.omit(windows[[stat]]) >= windows[[stat]][window])/length(na.omit(windows[[stat]]))\n",
    "                \n",
    "                result <- c(windows[[\"pop\"]][window], windows[[\"ID\"]][window], windows[[stat]][window], \n",
    "                  pvalue)\n",
    "                statPvalue <- rbind(statPvalue, result)\n",
    "                \n",
    "            }\n",
    "            print(result)\n",
    "        }\n",
    "        statPvalue <- as.data.frame(statPvalue)\n",
    "        colnames(statPvalue) <- c(\"pop\", \"ID\", stat, \"Pvalue\")\n",
    "        statPvalue[[\"adjust\"]] <- p.adjust(as.numeric(as.character(statPvalue[[\"Pvalue\"]])), \n",
    "            \"fdr\")\n",
    "        row.names(statPvalue) <- NULL\n",
    "    }\n",
    "    \n",
    "    return(statPvalue)\n",
    "}  # for xpehh, fst and ihs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if(!exists('genes2tad2windows')){\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData\")\n",
    "}\n",
    "pops<-c(\"YRI\",\"ACB\",\"LWK\",\"ESN\",\"GWD\",\"MSL\",\"ASW\",\"CEU\",\"GBR\",\"FIN\",\"IBS\",\"TSI\",\"CDX\",\"CHB\",\"CHS\",\"JPT\",\"KHV\",\"BEB\",\"GIH\",\"ITU\",\"PJL\",\"STU\")\n",
    "pops <- c('YRI','ACB')\n",
    "metapops<-c(\"CEU2CHB\",\"CEU2YRI\",\"YRI2CHB\")\n",
    "metapops<-c(\"CEU2CHB\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# Tajima_D\n",
    "Tajima_D <- list()\n",
    "for (i in pops){\n",
    "  Tajima_D[[i]]<-pvalueEmpiricV1(pop = i,stat = 'Tajima_D',autosomes=T,sexualX=T,sexualY=T)\n",
    "}\n",
    "Tajima_D <- do.call('rbind',Tajima_D)\n",
    "save(Tajima_D,file='~/pophuman.uab.cat/Results/10kb/Pvalues/Tajima_DPvalue.RData')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#iHS \n",
    "iHS<-list()\n",
    "for (i in pops){\n",
    "  iHS[[i]] <- pvalueEmpiricV3(pop = i,stat = 'iHS'autosomes=T,sexualX=T,sexualY=T)\n",
    "}\n",
    "iHS <- do.call('rbind',iHS)\n",
    "save(iHS,file='~/pophuman.uab.cat/Results/10kb/Pvalues/iHSPvalue.RData')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#FayWu_H \n",
    "FayWu_H<-list()\n",
    "for (i in pops){\n",
    "  FayWu_H[[i]] <- pvalueEmpiricV1(pop = i,stat = 'FayWu_H',autosomes=T,sexualX=T,sexualY=T)\n",
    "}\n",
    "FayWu_H <- do.call('rbind',FayWu_H)\n",
    "save(FayWu_H,file='~/pophuman.uab.cat/Results/10kb/Pvalues/FayWu_HPvalue.RData')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#FuLi_D \n",
    "FuLi_D<-list()\n",
    "for (i in pops){\n",
    "  FuLi_D[[i]]<-pvalueEmpiricV1(pop = i,stat = 'FuLi_D',autosomes=T,sexualX=T,sexualY=T)\n",
    "}\n",
    "FuLi_D <- do.call('rbind',FuLi_D)\n",
    "save(FuLi_D,file='~/pophuman.uab.cat/Results/10kb/Pvalues/FuLi_DPvalue.RData')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#FuLi_F \n",
    "FuLi_F<-list()\n",
    "for (i in pops){\n",
    "  FuLi_F[[i]]<-pvalueEmpiricV1(pop = i,stat = 'FuLi_F',autosomes=T,sexualX=T,sexualY=T)\n",
    "}\n",
    "FuLi_F <- do.call('rbind',FuLi_F)\n",
    "save(FuLi_F,file='~/pophuman.uab.cat/Results/10kb/Pvalues/FuLi_FPvalue.RData')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#theta\n",
    "theta<-list()\n",
    "for (i in pops){\n",
    "  theta[[i]]<-pvalueEmpiricV2(pop = i,stat = 'theta',autosomes=T,sexualX=T,sexualY=T)\n",
    "}\n",
    "theta <- do.call('rbind',theta)\n",
    "save(theta,file='~/pophuman.uab.cat/Results/10kb/Pvalues/thetaPvalue.RData')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Pi \n",
    "Pi<-list()\n",
    "for (i in pops){\n",
    "  Pi[[i]]<-pvalueEmpiricV2(pop = i,stat = 'Pi',autosomes=T,sexualX=T,sexualY=T)\n",
    "}\n",
    "Pi <- do.call('rbind',Pi)\n",
    "save(Pi,file='~/pophuman.uab.cat/Results/10kb/Pvalues/PiPvalue.RData')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#XPEHH \n",
    "XPEHH<-list()\n",
    "for (i in metapops){\n",
    "  XPEHH[[i]]<-pvalueEmpiricV3(pop = i,stat = 'XPEHH'autosomes=T,sexualX=T,sexualY=T)}\n",
    "XPEHH <- do.call('rbind',XPEHH)\n",
    "save(XPEHH,file='~/pophuman.uab.cat/Results/10kb/Pvalues/XPEHHPvalue.RData')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-23T13:55:50.478159Z",
     "start_time": "2018-05-23T13:58:25.158Z"
    },
    "code_folding": [
     0
    ],
    "hidden": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#Fst \n",
    "Fst<-list()\n",
    "for (i in metapops){\n",
    "  Fst[[i]]<-pvalueEmpiricV3(pop = i,stat = 'Fst',autosome=T,sexualX=T,sexualY=F)\n",
    "}\n",
    "Fst <- do.call('rbind',Fst)\n",
    "save(Fst,file='~/pophuman.uab.cat/Results/10kb/Pvalues/FstPvalue.RData')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Saving significatives data.frames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [
     1
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "load('~/pophuman.uab.cat/Results/10kb/contigousWindows.RData')\n",
    "savingSignificatives <- function(statistic,cutoff){\n",
    "    # VARIABLES TO SAVE RESULTS\n",
    "    Tajima_D_Corrected <- data.frame()\n",
    "    FayWu_H_Corrected <- data.frame()\n",
    "    FuLi_D_Corrected <- data.frame()\n",
    "    FuLi_F_Corrected <- data.frame()\n",
    "    Pi_Corrected <- data.frame()\n",
    "    theta_Corrected <- data.frame()\n",
    "    iHS_Corrected <- data.frame()\n",
    "    XPEHH_Corrected <- data.frame()\n",
    "    Fst_Corrected <- data.frame()\n",
    "\n",
    "    print('+++++++++')\n",
    "    print(paste0('loading: ',statistic))\n",
    "    print('+++++++++')\n",
    "    var <- paste0('~/pophuman.uab.cat/Results/10kb/Pvalues/',statistic,'Pvalue.RData')\n",
    "    varEnvir <- get(load(var))\n",
    "\n",
    "    varEnvir[['Pvalue']] <- as.numeric(as.character(varEnvir[['Pvalue']]))\n",
    "    varEnvir[['pop']] <- as.character(varEnvir[['pop']])\n",
    "    varEnvir[['ID']] <- as.numeric(as.character(varEnvir[['ID']]))\n",
    "    varEnvir[[statistic]] <- as.numeric(as.character(varEnvir[[statistic]]))\n",
    "        \n",
    "    if(statistic == 'Tajima_D' || statistic == 'FayWu_H' || statistic == 'FuLi_D' ||  statistic == 'FuLi_F'){\n",
    "            varEnvir <- varEnvir[varEnvir[['Pvalue']]*2 <= cutoff ,]\n",
    "    }\n",
    "    varEnvir <- varEnvir[varEnvir[['Pvalue']] <= cutoff ,]\n",
    "\n",
    "    varEnvir <- merge(varEnvir,contigousWindows,by = 'ID')\n",
    "    varEnvir <- varEnvir[,c('ID','pop','Pvalue',statistic,'chr','Group')]\n",
    "\n",
    "    if( statistic == 'Tajima_D'){\n",
    "          Tajima_D_Corrected <- varEnvir\n",
    "          row.names(Tajima_D_Corrected) <- NULL\n",
    "          return(Tajima_D_Corrected)\n",
    "        }\n",
    "    if( statistic == 'FayWu_H'){\n",
    "          FayWu_H_Corrected <- varEnvir\n",
    "          row.names(FayWu_H_Corrected) <- NULL\n",
    "          return(FayWu_H_Corrected)\n",
    "        }\n",
    "    if( statistic == 'FuLi_D'){\n",
    "          FuLi_D_Corrected <- varEnvir\n",
    "          row.names(FuLi_D_Corrected) <- NULL\n",
    "          return(FuLi_D_Corrected)\n",
    "        }\n",
    "    if( statistic == 'FuLi_F'){\n",
    "          FuLi_F_Corrected <- varEnvir\n",
    "          row.names(FuLi_F_Corrected) <- NULL\n",
    "          return(FuLi_F_Corrected)\n",
    "        }\n",
    "    if( statistic == 'theta'){\n",
    "          theta_Corrected <- varEnvir\n",
    "            row.names(theta_Corrected) <- NULL\n",
    "\n",
    "          return(theta_Corrected)}\n",
    "    if( statistic == 'Pi'){\n",
    "          Pi_Corrected <- varEnvir\n",
    "            row.names(Pi_Corrected) <- NULL\n",
    "\n",
    "          return(Pi_Corrected)}\n",
    "    if( statistic == 'iHS'){\n",
    "          iHS_Corrected <- varEnvir\n",
    "            row.names(iHS_Corrected) <- NULL\n",
    "      return(iHS_Corrected)}\n",
    "    if( statistic == 'XPEHH'){\n",
    "          XPEHH_Corrected <- varEnvir\n",
    "            row.names(XPEHH_Corrected) <- NULL\n",
    "        \n",
    "          return(XPEHH_Corrected)}\n",
    "    if( statistic == 'Fst'){\n",
    "          Fst_Corrected <- varEnvir\n",
    "             row.names(Fst_Corrected) <- NULL\n",
    "\n",
    "          return(Fst_Corrected)}\n",
    "    \n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dfTajima_DCorrected <- savingSignificatives('Tajima_D',cutoff=0.0005)\n",
    "save(dfTajima_DCorrected, file='~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/Tajima_DPvalueSig.RData')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dfiHSCorrected <- savingSignificatives('iHS',cutoff=0.005) \n",
    "save(dfiHSCorrected, file='~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/iHSPvalueSig.RData')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dfFstCorrected <- savingSignificatives('Fst',cutoff=0.0005) \n",
    "save(dfFstCorrected, file='~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/FstPvalueSig.RData')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dfXPEHHCorrected <- savingSignificatives('XPEHH',cutoff=0.0005)\n",
    "save(dfXPEHHCorrected, file='~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/XPEHHPvalueSig.RData')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dfFayWu_HCorrected <- savingSignificatives('FayWu_H',cutoff=0.0005) \n",
    "save(dfFayWu_HCorrected, file='~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/FayWu_HPvalueSig.RData')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dfFuLi_FCorrected <- savingSignificatives('FuLi_F',cutoff=0.0005) \n",
    "save(dfFuLi_FCorrected, file='~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/FuLi_FPvalueSig.RData')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dfFuLi_DCorrected <- savingSignificatives('FuLi_D',cutoff=0.0005) \n",
    "save(dfFuLi_DCorrected, file='~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/FuLi_DPvalueSig.RData')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-28T08:37:47.964215Z",
     "start_time": "2018-05-28T08:38:45.920Z"
    },
    "code_folding": [],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dfPiCorrected <-  savingSignificatives('Pi',cutoff=0.0005) \n",
    "save(dfPiCorrected, file='~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/PiPvalueSig.RData')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dfthetaCorrected <-  savingSignificatives('theta',cutoff=0.0005) \n",
    "save(dfthetaCorrected, file='~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/thetaPvalueSig.RData')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Descriptive data features"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Statistics mean and standard deviations by population"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-30T11:01:23.934978Z",
     "start_time": "2018-05-30T11:04:50.710Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if(!exists('genes2tad2windows')){\n",
    "    load('~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData')\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-24T11:22:56.325381Z",
     "start_time": "2018-05-24T11:26:24.915Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "summary <- genes2tad2windows[genes2tad2windows[['pop']]!='CEU2CHB' & genes2tad2windows[['pop']]!='CEU2YRI'  & genes2tad2windows[['pop']]!='YRI2CHB',] %>% group_by(pop) %>% summarise(windows=length(pop),mean_S=mean(na.omit(S)),sd_S=sd(na.omit(S)),mean_theta=mean(na.omit(theta)),sd_theta=sd(na.omit(theta)),mean_pi=mean(na.omit(Pi)),sd_pi=sd(na.omit(Pi)),mean_Tajima_D=mean(na.omit(Tajima_D)),sd_Tajima_D=sd(na.omit(Tajima_D)),mean_FayWu_H=mean(na.omit(FayWu_H)),sd_FayWu_H=sd(na.omit(FayWu_H)),mean_FuLi_D=mean(na.omit(FuLi_D)),sd_FuLi_D=sd(na.omit(FuLi_D)),mean_FuLi_F=mean(na.omit(FuLi_F)),sd_FuLi_F=sd(na.omit(FuLi_F)),mean_iHS=mean(na.omit(iHS)),sd_iHS=sd(na.omit(iHS)))\n",
    "fwrite(summary,file='~/pophuman.uab.cat/Results/10kb/Features/meanSdTable.tab',sep='\\t',quote = F,col.names = T,row.names = F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-24T11:19:29.614727Z",
     "start_time": "2018-05-24T11:23:09.138Z"
    },
    "code_folding": [],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "summary3pop <- genes2tad2windows[genes2tad2windows[['pop']]=='CEU2CHB' | genes2tad2windows[['pop']]=='CEU2YRI'  | genes2tad2windows[['pop']]=='YRI2CHB',] %>% group_by(pop) %>% summarise(\n",
    "            windows=length(pop),\n",
    "            mean_XPEHH=mean(na.omit(XPEHH)),\n",
    "            sd_XPEHH=mean(na.omit(XPEHH)),\n",
    "            mean_Fst=mean(na.omit(Fst)),\n",
    "            sd_Fst=sd(na.omit(Fst)))\n",
    "\n",
    "fwrite(summary3pop,file='~/pophuman.uab.cat/Results/10kb/Features/meanSdTablePairwisePop.tab',sep='\\t',quote = F,col.names = T,row.names = F)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Number of genes by windows and populations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "#### 10kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-24T11:21:51.148423Z",
     "start_time": "2018-05-24T11:25:22.846Z"
    },
    "code_folding": [],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# Global variables\n",
    "if(!exists('genes2tad2windows')){\n",
    "    load('~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData')\n",
    "\n",
    "}\n",
    "\n",
    "pops<-c('ACB','ASW','BEB','CDX','CEU','CHB','CHS','ESN','FIN','GBR','GIH','GWD','IBS','ITU','JPT','KHV','LWK','MSL','PJL','STU','TSI','YRI')\n",
    "chr <-c(1:22,'X','Y')\n",
    "\n",
    "# Number of genes by pop\n",
    "temporalVector <- NULL\n",
    "genesNumberPop <- NULL\n",
    "for (i in pops){\n",
    "    temporalVector <- c(dim(genes2tad2windows[genes2tad2windows[['pop']] == i & !is.na(genes2tad2windows[['GeneID']]),c('ID','chr','GeneID')])[1],i)\n",
    "    genesNumberPop <- rbind(genesNumberPop, temporalVector)\n",
    "}\n",
    "\n",
    "genesNumberPop <- as.data.frame(genesNumberPop); row.names(genesNumberPop) <- NULL; colnames(genesNumberPop) <- c('genesNumber', 'Pop'); genesNumberPop[['genesNumber']] <- as.numeric(as.character(genesNumberPop[['genesNumber']]))\n",
    "\n",
    "# Number of windows by pop\n",
    "temporalVector <- NULL\n",
    "windowsNumberPop <- NULL\n",
    "for (i in pops){\n",
    "    temporalVector <- c(dim(genes2tad2windows[genes2tad2windows[['pop']] == i, c(1,2,16)])[1],i)\n",
    "    windowsNumberPop <- rbind(windowsNumberPop, temporalVector)\n",
    "}\n",
    "\n",
    "windowsNumberPop <- as.data.frame(windowsNumberPop); row.names(windowsNumberPop) <- NULL; colnames(windowsNumberPop) <- c('windowsNumber', 'Pop'); genesNumberPop[['genesNumber']] <- as.numeric(as.character(genesNumberPop[['genesNumber']]))\n",
    "\n",
    "mainWindowsFeatures <- merge(windowsNumberPop,genesNumberPop,by='Pop',all=T)\n",
    "\n",
    "fwrite(mainWindowsFeatures,'~/pophuman.uab.cat/Results/10kb/Features/mainWindowsFeatures.tab', quote = F,col.names = T,row.names = F,sep = '\\t') \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "#### 100-kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-24T11:23:12.775261Z",
     "start_time": "2018-05-24T11:26:53.045Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if(!exists('windows100kb') | !exists('mainWindowsFeatures')){\n",
    "    load('~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData')\n",
    "    mainWindowsFeatures <- fread('~/pophuman.uab.cat/Results/10kb/Features/mainWindowsFeatures.tab')\n",
    "}\n",
    "\n",
    "# Global variables\n",
    "pops<-c('ACB','ASW','BEB','CDX','CEU','CHB','CHS','ESN','FIN','GBR','GIH','GWD','IBS','ITU','JPT','KHV','LWK','MSL','PJL','STU','TSI','YRI')\n",
    "chr <-c(1:22,'X','Y')\n",
    "\n",
    "# Number of windows by pop\n",
    "temporalVector <- NULL\n",
    "windowsNumberPop100kb <- NULL\n",
    "for (i in pops){\n",
    "    temporalVector <- c(dim(windows100kb[windows100kb[['pop']] == i, c('ID','chr')])[1],i)\n",
    "    windowsNumberPop100kb <- rbind(windowsNumberPop100kb, temporalVector)\n",
    "}\n",
    "\n",
    "windowsNumberPop100kb <- as.data.frame(windowsNumberPop100kb); row.names(windowsNumberPop100kb) <- NULL; colnames(windowsNumberPop100kb) <- c('windowsNumber100kb', 'Pop')\n",
    "\n",
    "mainWindowsFeatures <- merge(windowsNumberPop100kb,mainWindowsFeatures,by='Pop',all=T)\n",
    "fwrite(mainWindowsFeatures,'~/pophuman.uab.cat/Results/10kb/Features/mainWindowsFeatures.tab', quote = F,col.names = T,row.names = F,sep = '\\t') \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Number of windows by statistics and populations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-24T11:24:55.016007Z",
     "start_time": "2018-05-24T11:27:21.452Z"
    },
    "code_folding": [
     0
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if(!exists('genes2tad2windows')){\n",
    "    load('~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData')\n",
    "}\n",
    "\n",
    "# Global variables\n",
    "windowsStat <- NULL\n",
    "stat<-c('Tajima_D','FayWu_H','FuLi_D','FuLi_F','iHS','S','theta','Pi','XPEHH','Fst')\n",
    "pops<-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\")\n",
    "\n",
    "for (i in stat){\n",
    "    print(i)\n",
    "    if(i=='Fst' | i=='XPEHH'){\n",
    "        pops<-c('CEU2CHB','CEU2YRI','YRI2CHB')\n",
    "    }\n",
    "    for (j in pops){\n",
    "        tempWindowsStat<-NULL\n",
    "        temporalVector <- length(genes2tad2windows[[i]][genes2tad2windows[['pop']] == j])\n",
    "        tempMax = max(na.omit(genes2tad2windows[[i]][genes2tad2windows[['pop']] == j]))\n",
    "        tempMin = min(na.omit(genes2tad2windows[[i]][genes2tad2windows[['pop']] == j]))\n",
    "        tempWindowsStat <- c(j,i,temporalVector,tempMax,tempMin)\n",
    "        windowsStat <- rbind(windowsStat,tempWindowsStat)\n",
    "    }\n",
    "}\n",
    "\n",
    "\n",
    "windowsStat <- as.data.frame(windowsStat);row.names(windowsStat) <- NULL; colnames(windowsStat) <- c('pop', 'stat','windows','max','min')\n",
    "fwrite(windowsStat,'~/pophuman.uab.cat/Results/10kb/Features/windowsStat.tab', quote = F,col.names = T,row.names = F,sep = '\\t') "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Total number of significant windows under selected cutoffs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-18T14:51:02.383746Z",
     "start_time": "2018-04-18T14:53:47.056Z"
    },
    "code_folding": [],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stat<-c('Tajima_D','FayWu_H','FuLi_D','FuLi_F','theta','Pi','iHS','XPEHH','Fst')\n",
    "\n",
    "pops<-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\")\n",
    "\n",
    "windowsSig <- NULL\n",
    "for (i in stat){\n",
    "    print(i)\n",
    "    if(i=='Fst' | i=='XPEHH'){\n",
    "        pops<-c('CEU2CHB','CEU2YRI','YRI2CHB')}\n",
    "    for (j in pops){\n",
    "        fileName <- paste0('~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/',i,'PvalueSig.RData')\n",
    "        file <- get(load(fileName))\n",
    "        temporalVector <- length(file[[i]][file[['pop']] == j])\n",
    "        \n",
    "        if(i == 'iHS' | i == 'XPEHH'){\n",
    "         tailPos <- min(na.omit(file[file[['pop']] == j & file[[i]] > 0,i]))\n",
    "         tailNeg <- 0\n",
    "        }\n",
    "        else if (i == 'theta' | i == 'Pi' | i == 'Fst'){\n",
    "            tailPos <- min(na.omit(file[file[['pop']] == j & file[[i]] > 0,i]))\n",
    "            tailNeg <- max(na.omit(file[file[['pop']] == j & file[[i]] > 0,i]))\n",
    "        }\n",
    "        else{\n",
    "            tailPos <- min(na.omit(file[file[['pop']] == j & file[[i]] > 0,i]))\n",
    "            tailNeg <- max(na.omit(file[file[['pop']] == j & file[[i]] < 0,i]))\n",
    "        }\n",
    "                                    \n",
    "       \n",
    "        tempWindowsStat <- c(j,i,temporalVector,tailPos,tailNeg)\n",
    "        windowsSig <- rbind(windowsSig,tempWindowsStat)\n",
    "    }\n",
    "}\n",
    "    \n",
    "windowsSig <- as.data.frame(windowsSig);row.names(windowsSig) <- NULL; colnames(windowsSig) <- c('pop', 'stat','windowsSig','tailPos','tailNeg')\n",
    "windowsSig\n",
    "fwrite(windowsSig,file='~/pophuman.uab.cat/Results/10kb/Features/windowsSig.tab',sep='\\t',quote=F,col.names=T,row.names=F)\n",
    "\n",
    "# Merging windowsStats table\n",
    "windowsStat <- fread('~/pophuman.uab.cat/Results/10kb/Features/windowsStat.tab', header = T)\n",
    "\n",
    "windowsSigValues <- merge(windowsStat,windowsSig,by = c('pop','stat'),all = T)\n",
    "windowsSigValues <- windowsSigValues[order(windowsSigValues[['stat']]),]\n",
    "windowsSigValues\n",
    "\n",
    "fwrite(windowsSigValues,'~/pophuman.uab.cat/Results/10kb/Features/windowsSigValues.tab',sep='\\t',quote=F,row.names=F,col.names=T)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Pvalues distribution tables (featuresTails)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-28T11:10:27.625353Z",
     "start_time": "2018-05-28T11:08:54.215Z"
    },
    "code_folding": [],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "significantValues <- function(statistic) {\n",
    "    print(statistic)\n",
    "    print(\"+++++++++++++++++++++++\")\n",
    "    tails <- data.frame()\n",
    "    genesTableMajor <- data.frame()\n",
    "    genesTableMinor <- data.frame()\n",
    "    \n",
    "    pops <- c(\"YRI\", \"LWK\", \"GWD\", \"MSL\", \"ESN\", \"ACB\", \"ASW\", \"CHB\", \"JPT\", \"CHS\", \n",
    "        \"CDX\", \"KHV\", \"CEU\", \"TSI\", \"FIN\", \"GBR\", \"IBS\", \"GIH\", \"PJL\", \"BEB\", \"STU\", \n",
    "        \"ITU\")\n",
    "    \n",
    "    print(\"Loading data\")\n",
    "    temp <- paste0(\"~/pophuman.uab.cat/Results/10kb/Pvalues/\", statistic, \"Pvalue.RData\")\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/genes2ID.RData\")\n",
    "    load(\"~/pophuman.uab.cat/Results/10kb/contigousWindows.RData\")\n",
    "    \n",
    "    file <- get(load(temp))\n",
    "    \n",
    "    pop2 <- c(\"YRI\", \"ACB\", \"LWK\", \"ESN\", \"GWD\", \"MSL\", \"ASW\", \"CEU\", \"GBR\", \"FIN\", \n",
    "        \"IBS\", \"TSI\", \"CDX\", \"CHB\", \"CHS\", \"JPT\", \"KHV\", \"BEB\", \"GIH\", \"ITU\", \"PJL\", \n",
    "        \"STU\")\n",
    "    \n",
    "    if (statistic == \"Fst\" || statistic == \"XPEHH\") {\n",
    "        pop2 <- c(\"CEU2CHB\", \"CEU2YRI\", \"YRI2CHB\")\n",
    "    }\n",
    "    for (i in pop2) {\n",
    "        file[[i]][[\"pop\"]] <- i\n",
    "    }\n",
    "    \n",
    "    print(\"+++++++++++++++++++++++\")\n",
    "    \n",
    "    file[[statistic]] <- as.numeric(as.character(file[[statistic]]))\n",
    "    file[[\"Pvalue\"]] <- as.numeric(as.character(file[[\"Pvalue\"]]))\n",
    "    if (statistic == \"Tajima_D\" || statistic == \"FayWu_H\" || statistic == \"FuLi_D\" || \n",
    "        statistic == \"FuLi_F\") {\n",
    "        file[[\"Pvalue\"]] <- file[[\"Pvalue\"]] * 2\n",
    "    }\n",
    "    fileCutoff <- file[file[[\"Pvalue\"]] <= 5e-04, ]\n",
    "    fileCutoff <- na.omit(fileCutoff)\n",
    "    \n",
    "    fileCutoff <- merge(fileCutoff, contigousWindows, by = \"ID\")\n",
    "    fileCutoff <- merge(fileCutoff, genes2ID, by = \"ID\")\n",
    "    \n",
    "    if (statistic == \"Fst\" || statistic == \"XPEHH\") {\n",
    "        pops <- c(\"CEU2CHB\", \"CEU2YRI\", \"YRI2CHB\")\n",
    "    }\n",
    "    for (i in pops) {\n",
    "        print(i)\n",
    "        \n",
    "        tmpWindows <- length(na.omit(file[file[[\"pop\"]] == i, \"Pvalue\"]))\n",
    "        if (statistic == \"Fst\" || statistic == \"XPEHH\" || statistic == \"iHS\") {\n",
    "            tailMinor <- NA\n",
    "            contigousMinor <- NA\n",
    "            minStatistic <- NA\n",
    "            genesMinor <- NA\n",
    "            \n",
    "            genesMinorTable <- NA\n",
    "        } else {\n",
    "            tailMinor <- length(fileCutoff[fileCutoff[[\"pop\"]] == i & fileCutoff[[statistic]] < \n",
    "                0, \"Pvalue\"])\n",
    "            contigousMinor <- length(unique(fileCutoff[[\"Group\"]][fileCutoff[[\"pop\"]] == \n",
    "                i & fileCutoff[[statistic]] < 0]))\n",
    "            minStatistic <- max(fileCutoff[[statistic]][fileCutoff[[statistic]] < \n",
    "                0 & fileCutoff[[\"pop\"]] == i])\n",
    "            genesMinor <- length(na.omit(unique(unlist(strsplit(unique(fileCutoff[[\"GeneID\"]][fileCutoff[[\"pop\"]] == \n",
    "                i & fileCutoff[[statistic]] < 0]), split = \",\")))))\n",
    "            \n",
    "            tempMinorTable <- as.data.frame(na.omit(unique(unlist(strsplit(unique(fileCutoff[[\"GeneID\"]][fileCutoff[[\"pop\"]] == \n",
    "                i & fileCutoff[[statistic]] < 0]), split = \", \")))))\n",
    "            tempMinorTable[[\"pop\"]] <- i\n",
    "            \n",
    "            genesTableMinor <- rbind(genesTableMinor, tempMinorTable)\n",
    "            \n",
    "        }\n",
    "        \n",
    "        tailMajor <- length(fileCutoff[fileCutoff[[\"pop\"]] == i & fileCutoff[[statistic]] > \n",
    "            0, \"Pvalue\"])\n",
    "        maxStatistic <- min(fileCutoff[[statistic]][fileCutoff[[statistic]] > 0 & \n",
    "            fileCutoff[[\"pop\"]] == i])\n",
    "        contigousMajor <- length(unique(fileCutoff[[\"Group\"]][fileCutoff[[\"pop\"]] == \n",
    "            i & fileCutoff[[statistic]] > 0]))\n",
    "        genesMajor <- length(na.omit(unique(unlist(strsplit(unique(fileCutoff[[\"GeneID\"]][fileCutoff[[\"pop\"]] == \n",
    "            i & fileCutoff[[statistic]] > 0]), split = \", \")))))\n",
    "        \n",
    "        tempMajorTable <- as.data.frame(na.omit(unique(unlist(strsplit(unique(fileCutoff[[\"GeneID\"]][fileCutoff[[\"pop\"]] == \n",
    "            i & fileCutoff[[statistic]] > 0]), split = \", \")))))\n",
    "        tempMajorTable[[\"pop\"]] <- i\n",
    "        genesTableMajor <- rbind(genesTableMajor, tempMajorTable)\n",
    "        \n",
    "        \n",
    "        temp <- cbind(i, statistic, tmpWindows, tailMinor, tailMajor, contigousMinor, \n",
    "            contigousMajor, minStatistic, maxStatistic, genesMinor, genesMajor)\n",
    "        tails <- rbind(tails, temp)\n",
    "        \n",
    "    }\n",
    "    \n",
    "    # write.table(genesTableMinor,file=paste0('~/pophuman.uab.cat/Genes-based\n",
    "    # analysis/GenesWindowsTable/minor/',statistic,'Genes.tab'),sep='\\t',quote=F,row.names\n",
    "    # = F,col.names = F)\n",
    "    # write.table(genesTableMajor,file=paste0('~/pophuman.uab.cat/Genes-based\n",
    "    # analysis/GenesWindowsTable/major/',statistic,'Genes.tab'),sep='\\t',quote=F,row.names\n",
    "    # = F,col.names = F)\n",
    "    \n",
    "    return(tails)\n",
    "}\n",
    "\n",
    "stat <- c(\"Tajima_D\", \"FayWu_H\", \"iHS\", \"Fst\", \"XPEHH\", \"FuLi_F\", \"FuLi_D\")\n",
    "featuresTails <- data.frame()\n",
    "\n",
    "for (i in stat) {\n",
    "    tmp <- significantValues(i)\n",
    "    featuresTails <- rbind(featuresTails, tmp)\n",
    "}\n",
    "\n",
    "colnames(featuresTails) <- c(\"pop\", \"statistic\", \"windows\", \"tailMinor\", \"tailMajor\", \n",
    "    \"contigousMinor\", \"contigousMajor\", \"minStatistic\", \"maxStatistic\", \"genesMinor\", \n",
    "    \"genesMajor\")\n",
    "\n",
    "save(featuresTails, file = \"~/pophuman.uab.cat/Results/10kb/Features/featuresTails.RData\")\n",
    "fwrite(featuresTails, file = \"~/pophuman.uab.cat/Results/10kb/Features/featuresTails.tab\", \n",
    "    sep = \"\\t\", quote = F, row.names = F, col.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Plots"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Boxplots by populations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "#### 10kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-24T13:18:49.544058Z",
     "start_time": "2018-05-24T13:16:40.692Z"
    },
    "code_folding": [
     15
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if(!exists('genes2tad2windows')){\n",
    "    load('/home/jmurga/~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData')\n",
    "}\n",
    "\n",
    "populationPlot <- function(statistic,limit=c('5%','95%')){\n",
    "  ####################COLORCODE####################\n",
    "  pops2colors <-c(\"CEU\",\"GBR\",\"FIN\",\"IBS\",\"TSI\",\"ESN\",\"GWD\",\"LWK\",\"MSL\",\"YRI\",\"ACB\",\"ASW\",\"CDX\",\"CHB\",\"CHS\",\"JPT\",\"KHV\",\"BEB\",\"GIH\",\"ITU\",\"PJL\",\"STU\",\"CLM\",\"MXL\",\"PEL\",\"PUR\",\"CEU2CHB\",\"YRI2CHB\",\"CEU2YRI\")\n",
    "  colors<-c( \"#f5ec05\", \"#f7f14a\", \"#faf56b\", \"#f5f287\", \"#f5f39f\", \"#faf8b9\",\"#fafabe\", \"#008f00\", \"#33b033\", \"#37b337\", \"#6dd66d\", \"#90de90\", \"#2d74b2\",\"#5691c4\", \"#6fa6d6\", \"#8ab9e3\",\"#acd1f2\",\"#8e3ea3\", \"#a965ba\", \"#c587d6\", \"#d5a3e3\", \"#d3a2de\", \"#ab3b35\", \"#ba5852\", \"#cc7570\", \"#de9d99\",\"#008f00\",\"#5691c4\",\"#f5ec05\")\n",
    "  \n",
    "  #################################################\n",
    "  print(statistic)\n",
    "  genes2tad2windows <- as.data.frame(genes2tad2windows)\n",
    "    \n",
    "    pops <-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\")\n",
    "\n",
    "  if(statistic == 'XPEHH' || statistic == 'Fst'){\n",
    "    pops <- c('CEU2CHB','CEU2YRI','YRI2CHB')\n",
    "  }\n",
    "  temp10kb <- genes2tad2windows[,c('pop',statistic)]\n",
    "  temp10kb[['pop']]<-factor(temp10kb[['pop']], levels = pops)\n",
    "  temp10kb <- na.omit(temp10kb)\n",
    "  \n",
    "  limitPlot <- quantile(temp10kb[[statistic]], c(0.05,0.1, 0.25, 0.50, 0.75, 0.90,0.95))[limit]\n",
    "\n",
    "  \n",
    "  p <- ggplot(temp10kb,aes(x=temp10kb[['pop']], y=temp10kb[[statistic]],fill=temp10kb[['pop']]),show.legend = F) +\n",
    "    geom_violin(show.legend = F)+\n",
    "    geom_boxplot(width=0.3, outlier.shape = NA, position=position_dodge(0.75), fill = \"white\")\n",
    "  if(statistic == 'XPEHH' || statistic == 'Fst'){\n",
    "    p <- p + \n",
    "      scale_fill_manual('Populations',breaks = c(\"CEU2CHB\",\"YRI2CHB\",\"CEU2YRI\"),values = c(\"#008f00\",\"#5691c4\",\"#f5ec05\"))+\n",
    "      labs(y=paste0(statistic,', y limits = c(', substr(paste0(limitPlot,collapse = \" \",sep=','),1,nchar(paste0(limitPlot,collapse = \" \",sep=','))-1),')'),x='Populations',title=statistic) +\n",
    "      ylim(limitPlot)+theme(axis.text.x = element_text(size=14,face='bold',vjust = 0.5, angle = 90))+theme_bw()}else{\n",
    "        p <- p + \n",
    "          scale_fill_manual('Populations',breaks = pops2colors,values = colors)+\n",
    "          labs(y=paste0(statistic,', y limits = c(', substr(paste0(limitPlot,collapse = \" \",sep=','),1,nchar(paste0(limitPlot,collapse = \" \",sep=','))-1),')'),x='Populations',title=statistic) +\n",
    "          ylim(limitPlot)+ theme(axis.text.x = element_text(size=14,face='bold',vjust = 0.5, angle = 90)) + theme_bw()\n",
    "      }\n",
    "}\n",
    "\n",
    "stat<-c('Tajima_D','FayWu_H','FuLi_D','FuLi_F','iHS','S','theta','Pi','XPEHH','Fst')\n",
    "\n",
    "for(i in stat){\n",
    "    ggsave(filename = paste0('~/pophuman.uab.cat/Results/Plots/Features/Populations/10kb/',i,'PopulationPlot.png'),plot=populationPlot(statistic = i),dpi = 300 , width = 15,height = 10)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "#### 100kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-24T13:22:43.840359Z",
     "start_time": "2018-05-24T13:25:06.482Z"
    },
    "code_folding": [],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if(!exists('windows100kb')){\n",
    "    load('~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData')\n",
    "}\n",
    "pops<-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\",\"MXL\",\"PEL\",\"PUR\",\"CLM\")\n",
    "stat<-c('Tajima_D','FayWu_H','FuLi_D','FuLi_F','S','theta','Pi','Fst')\n",
    "\n",
    "populationPlot100kb <- function(statistic,limit=c('5%','95%')){\n",
    "  ####################COLORCODE####################\n",
    "  pops2colors <-c(\"CEU\",\"GBR\",\"FIN\",\"IBS\",\"TSI\",\"ESN\",\"GWD\",\"LWK\",\"MSL\",\"YRI\",\"ACB\",\"ASW\",\"CDX\",\"CHB\",\"CHS\",\"JPT\",\"KHV\",\"BEB\",\"GIH\",\"ITU\",\"PJL\",\"STU\",\"CLM\",\"MXL\",\"PEL\",\"PUR\",\"CEU2CHB\",\"YRI2CHB\",\"CEU2YRI\")\n",
    "  colors<-c( \"#f5ec05\", \"#f7f14a\", \"#faf56b\", \"#f5f287\", \"#f5f39f\", \"#faf8b9\",\"#fafabe\", \"#008f00\", \"#33b033\", \"#37b337\", \"#6dd66d\", \"#90de90\", \"#2d74b2\",\"#5691c4\", \"#6fa6d6\", \"#8ab9e3\",\"#acd1f2\",\"#8e3ea3\", \"#a965ba\", \"#c587d6\", \"#d5a3e3\", \"#d3a2de\", \"#ab3b35\", \"#ba5852\", \"#cc7570\", \"#de9d99\",\"#008f00\",\"#5691c4\",\"#f5ec05\")\n",
    "  \n",
    "  #################################################\n",
    "  print(statistic)\n",
    "  windows100kb <- as.data.frame(windows100kb)\n",
    "\n",
    "  pops <-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\")#NEED THIS ORDER TO PRINT\n",
    "    \n",
    "  if(statistic == 'XPEHH' || statistic == 'Fst'){\n",
    "    pops <- c('CEU2CHB','CEU2YRI','YRI2CHB')\n",
    "  }\n",
    "  temp100kb <- windows100kb[,c('pop',statistic)]\n",
    "  temp100kb[['pop']]<-factor(temp100kb[['pop']], levels = pops)\n",
    "  temp100kb <- na.omit(temp100kb)\n",
    "  \n",
    "  limitPlot <- quantile(temp100kb[[statistic]], c(0.05,0.1, 0.25, 0.50, 0.75, 0.90,0.95))[limit]\n",
    "  \n",
    "  p <- ggplot(temp100kb,aes(x=temp100kb[['pop']], y=temp100kb[[statistic]],fill=temp100kb[['pop']]),show.legend = F) +\n",
    "    geom_violin(show.legend = F)+\n",
    "    geom_boxplot(width=0.3, outlier.shape = NA, position=position_dodge(0.75), fill = \"white\")\n",
    "  if(statistic == 'XPEHH' || statistic == 'Fst'){\n",
    "    p <- p + \n",
    "      scale_fill_manual('Populations',breaks = c(\"CEU2CHB\",\"YRI2CHB\",\"CEU2YRI\"),values = c(\"#008f00\",\"#5691c4\",\"#f5ec05\"))+\n",
    "      labs(y=paste0(statistic,', y limits = c(', substr(paste0(limitPlot,collapse = \" \",sep=','),1,nchar(paste0(limitPlot,collapse = \" \",sep=','))-1),')'),x='Populations',title=statistic) +\n",
    "      ylim(limitPlot)+theme(axis.text.x = element_text(size=14,face='bold',vjust = 0.5, angle = 90))+theme_bw()}else{\n",
    "        p <- p + \n",
    "          scale_fill_manual('Populations',breaks = pops2colors,values = colors)+\n",
    "          labs(y=paste0(statistic,', y limits = c(', substr(paste0(limitPlot,collapse = \" \",sep=','),1,nchar(paste0(limitPlot,collapse = \" \",sep=','))-1),')'),x='Populations',title=statistic) +\n",
    "          ylim(limitPlot)+ theme(axis.text.x = element_text(size=14,face='bold',vjust = 0.5, angle = 90)) + theme_bw()\n",
    "      }\n",
    "}\n",
    "\n",
    "for(i in stat){\n",
    "    ggsave(filename = paste0('~/pophuman.uab.cat/Results/Plots/Features/Populations/100kb/',i,'PopulationPlot.png'),plot=populationPlot100kb(statistic = i),dpi = 300 , width = 15,height = 10)\n",
    "}\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "#### Boxplot 10kb~100kb distributions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-24T14:12:39.340431Z",
     "start_time": "2018-05-24T14:11:17.781Z"
    },
    "code_folding": [
     0,
     5
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if(!exists('genes2tad2windows') | !exists('windows100kb')){\n",
    "  load('/home/jmurga/~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData')\n",
    "  load('~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData')\n",
    "}\n",
    "\n",
    "populationPlot10100 <- function(statistic,limit=c('5%','95%'),S=F){\n",
    "  \n",
    "  print(statistic)\n",
    "  genes2tad2windows <- as.data.frame(genes2tad2windows)\n",
    "  windows100kb <- as.data.frame(windows100kb)\n",
    "  pops<-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\")\n",
    "\n",
    "  if(statistic == 'XPEHH' || statistic == 'Fst'){\n",
    "    pops <- c('CEU2CHB','CEU2YRI','YRI2CHB')\n",
    "  }\n",
    "  \n",
    "  \n",
    "  temp10kb <- genes2tad2windows[,c('pop',statistic)]\n",
    "  temp10kb[['pop']]<-factor(temp10kb[['pop']], levels = pops)\n",
    "  temp10kb <- na.omit(temp10kb)\n",
    "  temp10kb <- melt(temp10kb, id.vars=c('pop'))\n",
    "  temp10kb[['new']] <- temp10kb[['pop']]\n",
    "  \n",
    "  temp100kb <- windows100kb[,c('pop',statistic)]\n",
    "  \n",
    "  if(statistic == 'XPEHH' || statistic == 'Fst'){\n",
    "    temp100kb <- temp100kb[temp100kb[['pop']] != 'CLM' | temp100kb[['pop']] != 'MXL' | temp100kb[['pop']] != 'PUR' | temp100kb[['pop']] != 'PEL',]\n",
    "    temp100kb <- na.omit(temp100kb)\n",
    "    temp100kb[['pop']]<-factor(temp100kb[['pop']], levels = pops)\n",
    "    temp100kb <- na.omit(temp100kb)\n",
    "    temp100kb <- melt(temp100kb, id.vars=c('pop'))\n",
    "    temp100kb[['new']] <- paste0(temp100kb[['pop']],'100')\n",
    "    temp100kb[['new']] <- as.factor(temp100kb[['new']])\n",
    "  }else{\n",
    "    temp100kb <- temp100kb[temp100kb[['pop']] != 'CLM' | temp100kb[['pop']] != 'MXL' | temp100kb[['pop']] != 'PUR' | temp100kb[['pop']] != 'PEL',]\n",
    "    temp100kb[['pop']]<-factor(temp100kb[['pop']], levels = pops)\n",
    "    temp100kb <- na.omit(temp100kb)\n",
    "    if(S == TRUE){\n",
    "      temp100kb[['S']] <- temp100kb[['S']]/10\n",
    "    }\n",
    "    temp100kb <- melt(temp100kb, id.vars=c('pop'))\n",
    "    temp100kb[['new']] <- factor(paste0(temp100kb[['pop']],'100'),levels = c(\"YRI100\",\"LWK100\",\"GWD100\",\"MSL100\",\"ESN100\",\"ACB100\",\"ASW100\",\"CHB100\",\"JPT100\",\"CHS100\",\"CDX100\",\"KHV100\",\"CEU100\",\"TSI100\",\"FIN100\",\"GBR100\",\"IBS100\",\"GIH100\",\"PJL100\",\"BEB100\",\"STU100\",\"ITU100\"))\n",
    "  }\n",
    "  \n",
    "  \n",
    "  tempPlot <- rbind(temp10kb,temp100kb)\n",
    "  limitPlot <- quantile(temp100kb[['value']], c(0.05,0.1, 0.25, 0.50, 0.75, 0.90,0.95))[limit]\n",
    "  p <- ggplot(tempPlot,aes(x=pop,y=value,fill=new))+\n",
    "    geom_boxplot(show.legend = F)\n",
    "  \n",
    "  if(statistic == 'XPEHH' || statistic == 'Fst'){\n",
    "    p <- p + \n",
    "      scale_fill_manual('Populations',breaks = c(\"CEU2CHB\",\"YRI2CHB\",\"CEU2YRI\",\"CEU2CHB100\",\"YRI2CHB100\",\"CEU2YRI100\"),values = c(\"#008f00\",\"#5691c4\",\"#f5ec05\",\"#008f00\",\"#5691c4\",\"#f5ec05\"))+\n",
    "      labs(y=paste0(statistic,', y limits = c(', substr(paste0(limitPlot,collapse = \" \",sep=','),1,nchar(paste0(limitPlot,collapse = \" \",sep=','))-1),')'),x='Populations',title=statistic) +\n",
    "      ylim(limitPlot)+theme(axis.text.x = element_text(size=14,face='bold',vjust = 0.5, angle = 90))+theme_bw()}\n",
    "  else{\n",
    "    p <- p + \n",
    "      scale_fill_manual('Populations',breaks = c('YRI','LWK','GWD','MSL','ESN','ACB','ASW','CHB','JPT','CHS','CDX','KHV','CEU','TSI','FIN','GBR','IBS','GIH','PJL','BEB','STU','ITU','YRI100','LWK100','GWD100','MSL100','ESN100','ACB100','ASW100','CHB100','JPT100','CHS100','CDX100','KHV100','CEU100','TSI100','FIN100','GBR100','IBS100','GIH100','PJL100','BEB100','STU100','ITU100'),values = c(\"#f5f39f\",\"#faf56b\",\"#f7f14a\",\"#f5f287\",\"#f5ec05\",\"#faf8b9\",\"#fafabe\",\"#33b033\",\"#6dd66d\",\"#37b337\",\"#008f00\",\"#90de90\",\"#2d74b2\",\"#acd1f2\",\"#6fa6d6\",\"#5691c4\",\"#8ab9e3\",\"#a965ba\",\"#d5a3e3\",\"#8e3ea3\",\"#d3a2de\",\"#c587d6\",\"#FFE7A1\",\"#FFE7A1\",\"#FFE7A1\",\"#FFE7A1\",\"#FFE7A1\",\"#FFE7A1\",\"#FFE7A1\",\"#39C439\",\"#39C439\",\"#39C439\",\"#39C439\",\"#39C439\",\"#005FBD\",\"#005FBD\",\"#005FBD\",\"#005FBD\",\"#005FBD\",\"#7F4C8C\",\"#7F4C8C\",\"#7F4C8C\",\"#7F4C8C\",\"#7F4C8C\"))+\n",
    "      labs(y=paste0(statistic,', y limits = c(', substr(paste0(limitPlot,collapse = \" \",sep=','),1,nchar(paste0(limitPlot,collapse = \" \",sep=','))-1),')'),x='Populations',title=statistic)+ylim(limitPlot)+theme(axis.text.x = element_text(size=14,face='bold',vjust = 0.5, angle = 90)) + theme_bw()\n",
    "  }\n",
    "}\n",
    "\n",
    "pops<-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\")\n",
    "\n",
    "stat<-c('Tajima_D','FayWu_H','FuLi_D','FuLi_F','S','theta','Pi','Fst')\n",
    "\n",
    "for(i in stat){\n",
    "  if(i != 'S'){\n",
    "    ggsave(filename = paste0('~/pophuman.uab.cat/Results/Plots/Features/Populations/10-100kb/',i,'PopulationPlot10100kb.png'),plot=populationPlot10100(statistic = i),dpi = 300 , width = 15,height = 10)\n",
    "\n",
    "  }\n",
    "  else{\n",
    "    ggsave(filename = paste0('~/pophuman.uab.cat/Results/Plots/Features/Populations/10-100kb/',i,'PopulationPlot10100kbnorm.png'),plot=populationPlot10100(statistic = i,S=T),dpi = 300 , width = 15,height = 10)\n",
    "  }\n",
    "}\n",
    "\n",
    "i <- 'S'\n",
    "ggsave(filename = paste0('~/pophuman.uab.cat/Results/Plots/Features/Populations/10-100kb/',i,'PopulationPlot10100kb.png'),plot=populationPlot10100(statistic = i),dpi = 300 , width = 15,height = 10)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Boxplots by chromosomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "#### 10kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-24T14:42:55.275986Z",
     "start_time": "2018-05-24T14:20:00.397Z"
    },
    "code_folding": [
     44
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if(!exists('genes2tad2windows')){\n",
    "    load('~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData')\n",
    "}\n",
    "\n",
    "chrPlot <- function(population,statistic,limit=c('5%','95%')){\n",
    "  ####################COLORCODE####################\n",
    "  popil <- as.data.frame(matrix(c(\"CEU\",\"GBR\",\"FIN\",\"IBS\",\"TSI\",\"ESN\",\"GWD\",\"LWK\",\"MSL\",\"YRI\",\"ACB\",\"ASW\",\"CDX\",\"CHB\",\"CHS\",\"JPT\",\"KHV\",\"BEB\",\"GIH\",\"ITU\",\"PJL\",\"STU\",\"CLM\",\"MXL\",\"PEL\",\"PUR\",\"CEU2CHB\",\"YRI2CHB\",\"CEU2YRI\"),ncol = 1))\n",
    "  mycols<-as.data.frame(matrix(c(\"#2d74b2\",\"#5691c4\", \"#6fa6d6\", \"#8ab9e3\",\"#acd1f2\", \"#f5ec05\", \"#f7f14a\", \"#faf56b\", \"#f5f287\", \"#f5f39f\", \"#faf8b9\",\"#fafabe\", \"#008f00\", \"#33b033\", \"#37b337\", \"#6dd66d\", \"#90de90\", \"#8e3ea3\", \"#a965ba\", \"#c587d6\", \"#d5a3e3\", \"#d3a2de\", \"#ab3b35\", \"#ba5852\", \"#cc7570\", \"#de9d99\",\"#008f00\",\"#5691c4\",\"#f5ec05\"),ncol=1))\n",
    "  colorCode<-as.data.frame(cbind(popil,mycols));colnames(colorCode)<-c('pop','color')\n",
    "  #################################################\n",
    "  genes2tad2windows <- as.data.frame(genes2tad2windows)\n",
    "  temp10kb <- genes2tad2windows[genes2tad2windows[['pop']] == population ,c('chr',statistic)]\n",
    "  temp10kb[['chr']] <- factor(temp10kb[['chr']], levels = chr)\n",
    "  temp10kb <- na.omit(temp10kb)\n",
    "  limitPlot <- quantile(temp10kb[[statistic]], c(0.05,0.1, 0.25, 0.50, 0.75, 0.90,0.95))[limit]\n",
    "    \n",
    "  if(statistic == 'XPEHH'){\n",
    "    p <- ggplot(temp10kb,aes(x=temp10kb[['chr']], y=temp10kb[[statistic]]),show.legend = FALSE) +\n",
    "      geom_violin(fill=colorCode$color[colorCode$pop == population]) +\n",
    "      geom_boxplot(width=0.1, outlier.shape = NA, position=position_dodge(0.75), fill = \"white\")\n",
    "    \n",
    "    p <- p + \n",
    "      labs(y=paste0(statistic,', y limits = c(', substr(paste0(limit,collapse = \" \",sep=','),1,nchar(paste0(limit,collapse = \" \",sep=','))-1),')'),x='Populations',title=population) +\n",
    "      ylim(limitPlot) +\n",
    "      theme_bw()\n",
    "  }else{\n",
    "    p <- ggplot(temp10kb,aes(x=temp10kb[['chr']], y=temp10kb[[statistic]]),show.legend = FALSE) +\n",
    "      geom_violin(fill=colorCode[['color']][colorCode[['pop']] == population]) +\n",
    "      geom_boxplot(width=0.2, outlier.shape = NA, position=position_dodge(0.75), fill = \"white\")\n",
    "    p <- p + \n",
    "      labs(y=paste0(statistic,', y limits = c(', substr(paste0(limitPlot,collapse = \" \",sep=','),1,nchar(paste0(limitPlot,collapse = \" \",sep=','))-1),')'),x='Populations',title=population) +\n",
    "      ylim(limitPlot) +\n",
    "      theme_bw()\n",
    "  }\n",
    "    return(p)}\n",
    "\n",
    "chr <- c(\"chr1\",\"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrY\",\"chrX\")\n",
    "pops<-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\",\"MXL\",\"PEL\",\"PUR\",\"CLM\")\n",
    "stat <- c('Tajima_D','FayWu_H','FuLi_D','FuLi_F','iHS','S','theta','Pi','XPEHH','Fst')\n",
    "\n",
    "for(i in stat){\n",
    "  print('+++++++++++++++++')\n",
    "  print(i)\n",
    "  print('+++++++++++++++++')\n",
    "  if(i == 'XPEHH' | i == 'Fst'){\n",
    "      pops <-c(\"CEU2YRI\",\"CEU2CHB\",\"YRI2CHB\")\n",
    "  }\n",
    "  for (j in pops){\n",
    "      ggsave(filename = paste0('~/pophuman.uab.cat/Results/Plots/Features/Chr/10kb/',i,'/',j,i,'ChrPlot.png'),plot=chrPlot(population = j, statistic = i),dpi = 300 , width = 15,height = 10)\n",
    "    }\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "#### 100kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-24T16:05:57.156867Z",
     "start_time": "2018-05-24T15:51:43.166Z"
    },
    "code_folding": [
     4
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if(!exists('windows100kb')){\n",
    "    load('~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData')\n",
    "}\n",
    "\n",
    "chrPlot100 <- function(population,statistic,limit=c('5%','95%')){\n",
    "  ####################COLORCODE####################\n",
    "  popil <- as.data.frame(matrix(c(\"CEU\",\"GBR\",\"FIN\",\"IBS\",\"TSI\",\"ESN\",\"GWD\",\"LWK\",\"MSL\",\"YRI\",\"ACB\",\"ASW\",\"CDX\",\"CHB\",\"CHS\",\"JPT\",\"KHV\",\"BEB\",\"GIH\",\"ITU\",\"PJL\",\"STU\",\"CLM\",\"MXL\",\"PEL\",\"PUR\",\"CEU2CHB\",\"YRI2CHB\",\"CEU2YRI\"),ncol = 1))\n",
    "  mycols<-as.data.frame(matrix(c(\"#2d74b2\",\"#5691c4\", \"#6fa6d6\", \"#8ab9e3\",\"#acd1f2\", \"#f5ec05\", \"#f7f14a\", \"#faf56b\", \"#f5f287\", \"#f5f39f\", \"#faf8b9\",\"#fafabe\", \"#008f00\", \"#33b033\", \"#37b337\", \"#6dd66d\", \"#90de90\", \"#8e3ea3\", \"#a965ba\", \"#c587d6\", \"#d5a3e3\", \"#d3a2de\", \"#ab3b35\", \"#ba5852\", \"#cc7570\", \"#de9d99\",\"#008f00\",\"#5691c4\",\"#f5ec05\"),ncol=1))\n",
    "  colorCode<-as.data.frame(cbind(popil,mycols));colnames(colorCode)<-c('pop','color')\n",
    "  #################################################\n",
    "  windows100kb <- as.data.frame(windows100kb)\n",
    "  temp100kb <- windows100kb[windows100kb[['pop']] == population ,c('chr',statistic)]\n",
    "  temp100kb[['chr']] <- factor(temp100kb[['chr']], levels = chr)\n",
    "  temp100kb <- na.omit(temp100kb)\n",
    "  limitPlot <- quantile(temp100kb[[statistic]], c(0.05,0.1, 0.25, 0.50, 0.75, 0.90,0.95))[limit]\n",
    "    \n",
    "  if(statistic == 'XPEHH'){\n",
    "    p <- ggplot(temp100kb,aes(x=temp100kb[['chr']], y=temp100kb[[statistic]]),show.legend = FALSE) +\n",
    "      geom_violin(fill=colorCode$color[colorCode$pop == population]) +\n",
    "      geom_boxplot(width=0.1, outlier.shape = NA, position=position_dodge(0.75), fill = \"white\")\n",
    "    \n",
    "    p <- p + \n",
    "      labs(y=paste0(statistic,', y limits = c(', substr(paste0(limit,collapse = \" \",sep=','),1,nchar(paste0(limit,collapse = \" \",sep=','))-1),')'),x='Populations',title=population) +\n",
    "      ylim(limitPlot)+\n",
    "      theme_bw()\n",
    "  }else{\n",
    "    p <- ggplot(temp100kb,aes(x=temp100kb[['chr']], y=temp100kb[[statistic]]),show.legend = FALSE) +\n",
    "      geom_violin(fill=colorCode[['color']][colorCode[['pop']] == population]) +\n",
    "      geom_boxplot(width=0.2, outlier.shape = NA, position=position_dodge(0.75), fill = \"white\")\n",
    "    p <- p + \n",
    "      labs(y=paste0(statistic,', y limits = c(', substr(paste0(limitPlot,collapse = \" \",sep=','),1,nchar(paste0(limitPlot,collapse = \" \",sep=','))-1),')'),x='Populations',title=population) +\n",
    "      ylim(limitPlot)+\n",
    "      theme_bw()\n",
    "  }\n",
    "    return(p)}\n",
    "\n",
    "chr <- c(\"chr1\",\"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrY\",\"chrX\")\n",
    "pops<-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\",\"MXL\",\"PEL\",\"PUR\",\"CLM\")\n",
    "stat <- c('Tajima_D','FayWu_H','FuLi_D','FuLi_F','S','theta','Pi','Fst')\n",
    "\n",
    "for(i in stat){\n",
    "  print('+++++++++++++++++')\n",
    "  print(i)\n",
    "  print('+++++++++++++++++')\n",
    "  if(i == 'XPEHH' | i == 'Fst'){\n",
    "      pops <-c(\"CEU2YRI\",\"CEU2CHB\",\"YRI2CHB\")\n",
    "  }\n",
    "  for (j in pops){\n",
    "      ggsave(filename = paste0('~/pophuman.uab.cat/Results/Plots/Features/Chr/100kb/',i,'/',j,i,'ChrPlot.png'),plot=chrPlot100(population = j, statistic = i),dpi = 300 , width = 15,height = 10)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "#### Boxplot 10kb~100kb distributions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-25T09:47:47.368480Z",
     "start_time": "2018-05-25T09:31:44.871Z"
    },
    "code_folding": [],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if(!exists('genes2tad2windows') | !exists('windows100kb')){\n",
    "    load('~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData')\n",
    "    load('~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData')\n",
    "}\n",
    "\n",
    "chrDistComparision <- function(population,statistic,limit=c('5%','95%'),violin=F){\n",
    "  ####################COLORCODE####################\n",
    "  popil <- as.data.frame(matrix(c('YRI','LWK','GWD','MSL','ESN','ACB','ASW','CHB','JPT','CHS','CDX','KHV','CEU','TSI','FIN','GBR','IBS','GIH','PJL','BEB','STU','ITU',\"CEU2CHB\",\"YRI2CHB\",\"CEU2YRI\",'YRI100','LWK100','GWD100','MSL100','ESN100','ACB100','ASW100','CHB100','JPT100','CHS100','CDX100','KHV100','CEU100','TSI100','FIN100','GBR100','IBS100','GIH100','PJL100','BEB100','STU100','ITU100',\"CEU2CHB100\",\"YRI2CHB100\",\"CEU2YRI100\"),ncol = 1))\n",
    "mycols<-as.data.frame(matrix(c(\"#f5f39f\",\"#faf56b\",\"#f7f14a\",\"#f5f287\",\"#f5ec05\",\"#faf8b9\",\"#fafabe\",\"#33b033\",\"#6dd66d\",\"#37b337\",\"#008f00\",\"#90de90\",\"#2d74b2\",\"#acd1f2\",\"#6fa6d6\",\"#5691c4\",\"#8ab9e3\",\"#a965ba\",\"#d5a3e3\",\"#8e3ea3\",\"#d3a2de\",\"#c587d6\",\"#008f00\",\"#5691c4\",\"#f5ec05\",\"#FFE7A1\",\"#FFE7A1\",\"#FFE7A1\",\"#FFE7A1\",\"#FFE7A1\",\"#FFE7A1\",\"#FFE7A1\",\"#39C439\",\"#39C439\",\"#39C439\",\"#39C439\",\"#39C439\",\"#005FBD\",\"#005FBD\",\"#005FBD\",\"#005FBD\",\"#005FBD\",\"#7F4C8C\",\"#7F4C8C\",\"#7F4C8C\",\"#7F4C8C\",\"#7F4C8C\",\"#008f00\",\"#5691c4\",\"#f5ec05\"),ncol=1))\n",
    "    \n",
    "  colorCode <- as.data.frame(cbind(popil,mycols));colnames(colorCode)<-c('pop','color')\n",
    "  colorCode[['color']] <- as.character(colorCode[['color']])\n",
    "  #################################################\n",
    "  \n",
    "  windows100kb <- as.data.frame(windows100kb)\n",
    "  colnames(windows100kb) <- c(\"ID\",\"pop\",\"chr\",\"startW\",\"endW\",\"Fst100\",\"Tajima_D100\",\"FayWu_H100\",\"FuLi_D100\",\"FuLi_F100\",\"S100\",\"theta100\",\"Pi100\",\"XPEHH100\")\n",
    "  temp100kb <- windows100kb[windows100kb[['pop']]==population,c('chr',paste0(statistic,'100'))]\n",
    "  temp100kb <- na.omit(temp100kb)\n",
    "  temp100kb <- melt(temp100kb, id.vars='chr')\n",
    "  \n",
    "  genes2tad2windows <- as.data.frame(genes2tad2windows)\n",
    "  temp10kb <- genes2tad2windows[genes2tad2windows[['pop']] == population ,c('chr',statistic)]\n",
    "  temp10kb[['chr']] <- factor(temp10kb[['chr']], levels = chr)\n",
    "  temp10kb <- na.omit(temp10kb)\n",
    "  temp10kb <- melt(temp10kb, id.vars='chr')\n",
    "  \n",
    "  tempPlot <-rbind(temp10kb,temp100kb)\n",
    "  limitPlot <- quantile(temp100kb[['value']], c(0.05,0.1, 0.25, 0.50, 0.75, 0.90,0.95))[limit]\n",
    "\n",
    "    if (violin == T){\n",
    "        p <- ggplot(tempPlot,aes(x=chr,y=value,fill=variable))+\n",
    "          geom_violin() +\n",
    "          scale_fill_manual(values = c(colorCode[colorCode[['pop']]==population,'color'],colorCode[colorCode[['pop']]==population,'color']))\n",
    "    \n",
    "        p <- p + \n",
    "          labs(y=paste0(statistic,', y limits = c(', substr(paste0(limitPlot,collapse = \" \",sep=','),1,nchar(paste0(limitPlot,collapse = \" \",sep=','))-1),')'),x='Populations',title=population) +\n",
    "          ylim(limitPlot) +\n",
    "      theme_bw()\n",
    "    }\n",
    "    else{\n",
    "        color100 <- paste0(population,'100')\n",
    "        p <- ggplot(tempPlot,aes(x=chr,y=value,fill=variable))+\n",
    "          geom_boxplot() +\n",
    "          scale_fill_manual(values = c(colorCode[colorCode[['pop']]==population,'color'],colorCode[colorCode[['pop']]==color100,'color']))\n",
    "    \n",
    "        p <- p + \n",
    "          labs(y=paste0(statistic,', y limits = c(', substr(paste0(limitPlot,collapse = \" \",sep=','),1,nchar(paste0(limitPlot,collapse = \" \",sep=','))-1),')'),x='Populations',title=population) +\n",
    "          ylim(limitPlot) +\n",
    "      theme_bw()\n",
    "    }\n",
    "    return(p)\n",
    "  }\n",
    "\n",
    "chr <- c(\"chr1\",\"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrY\",\"chrX\")\n",
    "\n",
    "pops<-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\")\n",
    "stat <- c('Tajima_D','FayWu_H','FuLi_D','FuLi_F','S','theta','Pi','Fst')\n",
    "\n",
    "for(i in stat){\n",
    "  print('+++++++++++++++++')\n",
    "  print(i)\n",
    "  print('+++++++++++++++++')\n",
    "  if(i == 'XPEHH' | i == 'Fst'){\n",
    "      pops <-c(\"CEU2YRI\",\"CEU2CHB\",\"YRI2CHB\")\n",
    "  }\n",
    "  for (j in pops){\n",
    "      ggsave(filename = paste0('~/pophuman.uab.cat/Results/Plots/Features/Chr/10-100kb/',i,'/',j,i,'ChrPlot.png'),plot=chrDistComparision(population = j, statistic = i),dpi = 300 , width = 15,height = 10)\n",
    "    }\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Ploting distributions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "#### 10kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-18T12:16:32.981002Z",
     "start_time": "2018-04-18T08:13:23.932Z"
    },
    "code_folding": [
     8
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if(!exists('genes2tad2windows')){\n",
    "    load('~/pophuman.uab.cat/Results/10kb/genes2tad2windows.RData')\n",
    "}\n",
    "\n",
    "GenomeWidePlot <- function(statistic){\n",
    "    genomewide_tests<-NULL\n",
    "    genes2tad2windows <- as.data.frame(genes2tad2windows)\n",
    "    pops<-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\",\"MXL\",\"PEL\",\"PUR\",\"CLM\")\n",
    "    if (statistic == \"set1\"){\n",
    "        for (stat in c('Tajima_D','FayWu_H','FuLi_F','FuLi_D','Pi','theta','S')){\n",
    "            print('++++++++++++++++')\n",
    "            print(stat)\n",
    "            print('++++++++++++++++')\n",
    "            for (pop in pops){\n",
    "                print(pop)\n",
    "                genomewidedistr <- genes2tad2windows[genes2tad2windows[['pop']]==pop,c(\"chr\",\"startW\",\"endW\",stat)];colnames(genomewidedistr)<-c(\"Chr\",\"Start\",\"End\",'Value')\n",
    "                genomewidedistr[['Chr']] <- factor(genomewidedistr[['Chr']],levels=paste0(\"chr\",c(seq(1,22),\"X\",\"Y\")),labels=c(seq(1,22),\"X\",\"Y\")) #Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "                genomewidedistr[['Population']] <- pop\n",
    "                genomewidedistr[['Test']] <- stat\n",
    "                genomewidedistr <- na.omit(genomewidedistr)\n",
    "                genomewidedistr[['Quantile']]<-cut(genomewidedistr[['Value']],breaks=quantile(genomewidedistr[['Value']],probs = c(0,0.005,0.025,0.975,0.995,1)),labels=c(\"<0.01_Low\",\"<0.05_Low\",\">0.05_Mid\",\"<0.05_Top\",\"<0.01_Top\"),include.lowest = TRUE)\n",
    "                genomewide_tests<-rbind(genomewide_tests,genomewidedistr)\n",
    "                genomewide_tests[['Population']] <- factor(genomewide_tests[['Population']],levels=c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\",\"MXL\",\"PEL\",\"PUR\",\"CLM\")) #Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "                ggsave(get.plot(genomewide_tests,stat,pop), file=paste0('~/pophuman.uab.cat/Results/Plots/Features/Distributions/10kb/',stat,'/',pop,'_',stat,'_','Distribution.png'),dpi=150,width = 12,height = 8)\n",
    "            }\n",
    "        }\n",
    "    }\n",
    "    else if(statistic == 'iHS'){\n",
    "        print('++++++++++++++++')\n",
    "        print(stat)\n",
    "        print('++++++++++++++++')\n",
    "        for (pop in pops){\n",
    "            print(pop)\n",
    "            genomewidedistr <- genes2tad2windows[genes2tad2windows[['pop']]==pop,c(\"chr\",\"startW\",\"endW\",statistic)];colnames(genomewidedistr)<- c (\"Chr\",\"Start\",\"End\",'Value')\n",
    "            genomewidedistr[['Chr']]<-factor(genomewidedistr[['Chr']],levels=paste0(\"chr\",c(seq(1,22),\"X\",\"Y\")),labels=c(seq(1,22),\"X\",\"Y\"))\n",
    "            genomewidedistr[['Population']]<-pop\n",
    "            genomewidedistr[['Test']] <- statistic\n",
    "            genomewidedistr <- na.omit(genomewidedistr)\n",
    "            \n",
    "            genomewidedistr[['Quantile']]<-factor(as.character(cut(genomewidedistr[['Value']],breaks=quantile(genomewidedistr$Value,probs = c(0,0.95,0.99,1)),labels=c(\">0.05_Mid\",\"<0.05_Top\",\"<0.01_Top\"),include.lowest = TRUE)),levels=c(\"<0.01_Low\",\"<0.05_Low\",\">0.05_Mid\",\"<0.05_Top\",\"<0.01_Top\")) #Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "            genomewide_tests <- rbind(genomewide_tests,genomewidedistr)\n",
    "            genomewide_tests[['Population']]<-factor(genomewide_tests[['Population']],levels=c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\",\"MXL\",\"PEL\",\"PUR\",\"CLM\")) #Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "            \n",
    "            ggsave(get.plot(genomewide_tests,statistic,pop),file=paste0('~/pophuman.uab.cat/Results/Plots/Features/Distributions/10kb/',statistic,'/',pop,'_',statistic,'_','Distribution.png'),dpi=150,width = 12,height = 8)      \n",
    "        }\n",
    "    }\n",
    "    else if(statistic == 'XPEHH' || statistic == 'Fst'){\n",
    "        print(statistic)\n",
    "        for (pop in c(\"CEU2CHB\",\"CEU2YRI\",\"YRI2CHB\")){\n",
    "            print(pop)\n",
    "            genomewidedistr<-genes2tad2windows[genes2tad2windows[['pop']]==pop,c(\"chr\",\"startW\",\"endW\",statistic)];colnames(genomewidedistr)<-c(\"Chr\",\"Start\",\"End\",'Value')\n",
    "            genomewidedistr[['Chr']]<-factor(genomewidedistr[['Chr']],levels=paste0(\"chr\",c(seq(1,22),\"X\",\"Y\")),labels=c(seq(1,22),\"X\",\"Y\"))#Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "            \n",
    "            genomewidedistr[['Population']] <- pop\n",
    "            genomewidedistr[['Test']] <- statistic\n",
    "            genomewidedistr <- na.omit(genomewidedistr)\n",
    "            \n",
    "            genomewidedistr[['Quantile']]<-factor(as.character(cut(genomewidedistr[['Value']],breaks=quantile(genomewidedistr$Value,probs = c(0,0.95,0.99,1)),labels=c(\">0.05_Mid\",\"<0.05_Top\",\"<0.01_Top\"),include.lowest = TRUE)),levels=c(\"<0.01_Low\",\"<0.05_Low\",\">0.05_Mid\",\"<0.05_Top\",\"<0.01_Top\")) #Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "            \n",
    "            genomewide_tests <- rbind(genomewide_tests,genomewidedistr)\n",
    "            genomewide_tests[['Population']]<-factor(genomewide_tests[['Population']],levels=c(\"CEU2CHB\",\"CEU2YRI\",\"YRI2CHB\"))#Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "            \n",
    "            ggsave(get.plot(genomewide_tests,statistic,pop),file=paste0('~/pophuman.uab.cat/Results/Plots/Features/Distributions/10kb/',statistic,'/',pop,'_',statistic,'_','Distribution.png'),dpi=150,width = 12,height = 8)\n",
    "        }\n",
    "    }\n",
    "}\n",
    "get.plot<-function(df,statistic,population){\n",
    "  require(ggplot2)\n",
    "  ptest<-ggplot(df[df[['Test']] %in% statistic & df[['Population']] == population,])+\n",
    "    geom_point(aes(Start,Value,colour=Quantile),alpha=0.3, show.legend=FALSE)+\n",
    "    facet_grid(Population~Chr,scales=\"free_x\",space=\"free_x\")\n",
    "  if(statistic == 'XPEHH' || statistic == 'iHS' || statistic == 'Fst'){\n",
    "    ptest<- ptest + scale_color_manual(values=c(\"#C4C4C4\",\"#C4C4C4\",\"#5AB4AC\"))\n",
    "  }\n",
    "  else{\n",
    "    ptest <- ptest + scale_color_manual(values=c(\"#A6611A\",\"#DFC27D\",\"#C4C4C4\" ,\"#80CDC1\",\"#018571\"))\n",
    "    \n",
    "  }\n",
    "  ptest<- ptest +\n",
    "    theme(panel.spacing.x = unit(1,\"pt\"),\n",
    "          axis.text.x = element_blank(),\n",
    "          axis.ticks.x = element_blank(),\n",
    "          axis.title.x = element_blank())+\n",
    "    labs(title=statistic)\n",
    "  return(ptest)\n",
    "}\n",
    "\n",
    "GenomeWidePlot(statistic = 'iHS')\n",
    "GenomeWidePlot(statistic = 'Fst')\n",
    "GenomeWidePlot(statistic = 'XPEHH')\n",
    "GenomeWidePlot(statistic = 'set1')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "#### 100kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-18T13:43:44.712476Z",
     "start_time": "2018-04-18T13:29:13.438Z"
    },
    "code_folding": [
     5,
     50
    ],
    "hidden": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "if(!exists('windows100kb')){\n",
    "    load('~/pophuman.uab.cat/Results/100kb/windowsStatisticsPopulations100kb.RData')\n",
    "}\n",
    "\n",
    "\n",
    "GenomeWidePlot100 <- function(statistic){\n",
    "  genomewide_tests<-NULL\n",
    "  windows100kb <- as.data.frame(windows100kb)\n",
    "    pops<-c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\",\"MXL\",\"PEL\",\"PUR\",\"CLM\")\n",
    "  if (statistic == \"set1\"){\n",
    "    for (stat in c('Tajima_D','FayWu_H','FuLi_F','FuLi_D','Pi','theta','S')){\n",
    "      print(stat)\n",
    "      for (pop in pops){\n",
    "        print(pop)\n",
    "        genomewidedistr <- windows100kb[windows100kb[['pop']]==pop,c(\"chr\",\"startW\",\"endW\",stat)];colnames(genomewidedistr)<-c(\"Chr\",\"Start\",\"End\",'Value')\n",
    "        genomewidedistr[['Chr']] <- factor(genomewidedistr[['Chr']],levels=paste0(\"chr\",c(seq(1,22),\"X\",\"Y\")),labels=c(seq(1,22),\"X\",\"Y\")) #Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "        \n",
    "        genomewidedistr[['Population']] <- pop\n",
    "        genomewidedistr[['Test']] <- stat\n",
    "        genomewidedistr <- na.omit(genomewidedistr)\n",
    "        \n",
    "        genomewidedistr[['Quantile']]<-cut(genomewidedistr[['Value']],breaks=quantile(genomewidedistr[['Value']],probs = c(0,0.005,0.025,0.975,0.995,1)),labels=c(\"<0.01_Low\",\"<0.05_Low\",\">0.05_Mid\",\"<0.05_Top\",\"<0.01_Top\"),include.lowest = TRUE)\n",
    "        \n",
    "        genomewide_tests<-rbind(genomewide_tests,genomewidedistr)\n",
    "        genomewide_tests[['Population']] <- factor(genomewide_tests[['Population']],levels=c(\"YRI\",\"LWK\",\"GWD\",\"MSL\",\"ESN\",\"ACB\",\"ASW\",\"CHB\",\"JPT\",\"CHS\",\"CDX\",\"KHV\",\"CEU\",\"TSI\",\"FIN\",\"GBR\",\"IBS\",\"GIH\",\"PJL\",\"BEB\",\"STU\",\"ITU\",\"MXL\",\"PEL\",\"PUR\",\"CLM\")) #Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "        \n",
    "        ggsave(get.plot(genomewide_tests,stat,pop), file=paste0('~/pophuman.uab.cat/Results/Plots/Features/Distributions/',stat,'/',pop,'_',stat,'_','Distribution.png'),dpi=150,width = 12,height = 8)\n",
    "      }\n",
    "    }\n",
    "  }\n",
    "  else if(statistic == 'Fst'){\n",
    "    print(statistic)\n",
    "    for (pop in c(\"CEU2CHB\",\"CEU2YRI\",\"YRI2CHB\")){\n",
    "      print(pop)\n",
    "      genomewidedistr<-windows100kb[windows100kb[['pop']]==pop,c(\"chr\",\"startW\",\"endW\",statistic)];colnames(genomewidedistr)<-c(\"Chr\",\"Start\",\"End\",'Value')\n",
    "      genomewidedistr[['Chr']]<-factor(genomewidedistr[['Chr']],levels=paste0(\"chr\",c(seq(1,22),\"X\",\"Y\")),labels=c(seq(1,22),\"X\",\"Y\"))#Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "      \n",
    "      genomewidedistr[['Population']] <- pop\n",
    "      genomewidedistr[['Test']] <- statistic\n",
    "      genomewidedistr <- na.omit(genomewidedistr)\n",
    "      \n",
    "      genomewidedistr[['Quantile']]<-factor(as.character(cut(genomewidedistr[['Value']],breaks=quantile(genomewidedistr$Value,probs = c(0,0.95,0.99,1)),labels=c(\">0.05_Mid\",\"<0.05_Top\",\"<0.01_Top\"),include.lowest = TRUE)),levels=c(\"<0.01_Low\",\"<0.05_Low\",\">0.05_Mid\",\"<0.05_Top\",\"<0.01_Top\")) #Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "      \n",
    "      genomewide_tests <- rbind(genomewide_tests,genomewidedistr)\n",
    "      genomewide_tests[['Population']]<-factor(genomewide_tests[['Population']],levels=c(\"CEU2CHB\",\"CEU2YRI\",\"YRI2CHB\"))#Cannot change to genomewidedistr[['chr']], data.frame and factor problem\n",
    "      \n",
    "      ggsave(get.plot(genomewide_tests,statistic,pop),file=paste0('~/pophuman.uab.cat/Results/Plots/Features/Distributions/',statistic,'/',pop,'_',statistic,'_','Distribution.png'),dpi=150,width = 12,height = 8)\n",
    "    }\n",
    "  }\n",
    "}\n",
    "get.plot<-function(df,statistic,population){\n",
    "  require(ggplot2)\n",
    "  # windowsRecomb <- get(load('~/pophuman.uab.cat/Results/10kb/windowsRecomb.RData'))\n",
    "  ptest<-ggplot(df[df[['Test']] %in% statistic & df[['Population']] == population,])+\n",
    "    geom_point(aes(Start,Value,colour=Quantile),alpha=0.3, show.legend=FALSE)+\n",
    "    facet_grid(Population~Chr,scales=\"free_x\",space=\"free_x\")\n",
    "  if(statistic == 'XPEHH' || statistic == 'iHS' || statistic == 'Fst'){\n",
    "    ptest<- ptest + scale_color_manual(values=c(\"#C4C4C4\",\"#C4C4C4\",\"#5AB4AC\"))\n",
    "  }else{\n",
    "    ptest <- ptest + scale_color_manual(values=c(\"#A6611A\",\"#DFC27D\",\"#C4C4C4\" ,\"#80CDC1\",\"#018571\"))\n",
    "    \n",
    "  }\n",
    "  ptest<- ptest +\n",
    "    theme(panel.spacing.x = unit(1,\"pt\"),\n",
    "          axis.text.x = element_blank(),\n",
    "          axis.ticks.x = element_blank(),\n",
    "          axis.title.x = element_blank())+\n",
    "    labs(title=statistic)\n",
    "  return(ptest)}\n",
    "\n",
    "GenomeWidePlot100(statistic = 'Fst')\n",
    "GenomeWidePlot100(statistic = 'set1')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "table2Melt <- fread('~/pophuman.uab.cat/Results/10kb/Features/table2Melt.tab');colnames(table2Melt) <- c('pop','Statistic','Value','GeneID','Group','ID','Tail')\n",
    "enhancersTable <- fread('~/pophuman.uab.cat/Results/10kb/enhancersTable.tab',sep='\\t',header=T)\n",
    "\n",
    "enhancersTable <- enhancersTable[,c('ID','IDEnhancers')]\n",
    "tableEnh <- merge(table2Melt[table2Melt[['GeneID']]==\"\",c('pop','Statistic','Value','Group','ID','Tail')],enhancersTable,by=c('ID'),all.x=T)\n",
    "\n",
    "tableEnh <- unique(tableEnh[!is.na(tableEnh[['ID']]),])\n",
    "\n",
    "\n",
    "tempTableEnh1 <- tableEnh %>% group_by(pop,Statistic) %>% summarise(ID=toString(unique(ID)),Group=toString(unique(Group)),Tail=toString(unique(Tail)),Value=max(Value),IDEnhancers=toString(IDEnhancers))\n",
    "\n",
    "tempTableEnh2 <- tempTableEnh1 %>% group_by(GeneID,Statistic) %>% summarize(extremeValue=round(Value[which.max(abs(Value))],3),label=paste(pop,round(Value,3),sep = '=',collapse = \"; \"),ID = paste(pop,ID,sep = '=',collapse = \"; \"), Group = paste(pop,Group,sep = '=',collapse = \"; \"),labelCoord=toString(unique(labelCoord)),Coords=toString(unique(Coords))) \n",
    "\n",
    "\n",
    "labelEnh <- tempTableEnh1 %>% group_by(GeneID) %>% summarize(label=toString(unique(label)),ID=toString(unique(ID)),Group=toString(unique(Group)),labelCoord=toString(unique(labelCoord)),Coords=toString(unique(Coords)))\n",
    "\n",
    "tempTableEnh3 <- dcast(data=tempTableGenes2, GeneID ~ Statistic, value.var = c('extremeValue'))\n",
    "\n",
    "tableEnh <- merge(tempTableEnh3,label)\n",
    "\n",
    "tableEnh[['labelCoord']] <- sapply(tableEnh[['labelCoord']],function(x)strsplit(x,split=\",\")[[1]] %>% unique %>% paste(collapse=\",\")) # Delete duplicated entries\n",
    "tableEnh[['Coords']] <- sapply(tableEnh[['Coords']],function(x)strsplit(x,split=\",\")[[1]] %>% unique %>% paste(collapse=\",\")) # Delete duplicated entries\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Recomb distribution"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "#### Recomb by chromosome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-01T08:56:05.145235Z",
     "start_time": "2018-06-01T08:59:44.090Z"
    },
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "load('~/pophuman.uab.cat/Results/100kb/windowsRecomb100kb.RData')\n",
    "h(windowsRecomb)\n",
    "windowsRecomb100kb <-as.data.frame(windowsRecomb)\n",
    "windowsRecomb100kb[['chr']] <- as.factor(windowsRecomb100kb[['chr']])\n",
    "\n",
    "windowsRecomb100kb <- windowsRecomb100kb[windowsRecomb100kb[['chr']]=='chr1' | windowsRecomb100kb[['chr']]=='chr2' | windowsRecomb100kb[['chr']]=='chr3' | windowsRecomb100kb[['chr']]=='chr4' | windowsRecomb100kb[['chr']]=='chr5' | windowsRecomb100kb[['chr']]=='chr6' | windowsRecomb100kb[['chr']]=='chr7' | windowsRecomb100kb[['chr']]=='chr8' | windowsRecomb100kb[['chr']]=='chr9' |  windowsRecomb100kb[['chr']]=='chr10' ,]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-01T08:56:14.942386Z",
     "start_time": "2018-06-01T08:59:47.071Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "ggplot(windowsRecomb100kb,aes(x=windowsRecomb100kb[['startW']],y=windowsRecomb100kb[['recomb']],group = 'recomb')) +\n",
    "    geom_line()+\n",
    "    geom_smooth()+\n",
    "    facet_wrap(~chr)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "#### Plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-28T10:28:50.056390Z",
     "start_time": "2018-05-28T10:31:52.119Z"
    },
    "code_folding": [
     1,
     55,
     56,
     97
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "\n",
    "distrRecombPlot <- function(statistic, ecdf = FALSE, resampling=FALSE){\n",
    "\n",
    "    ####################COLORCODE####################\n",
    "    stat2colors <-c('globalRecomb',\"Tajima_D\",\"FayWu_H\",\"FuLi_F\",\"FuLi_D\",\"S\",\"Pi\",\"theta\",\"iHS\",\"XPEHH\",\"Fst\")\n",
    "    colors<-c('#FF4340',\"#f5ec05\",\"#f5f39f\",\"#2d74b2\",\"#6fa6d6\",\"#8e3ea3\", \"#a965ba\", \"#c587d6\",\"#d3a2de\", \"#ab3b35\", \"#008f00\")\n",
    "     #################################################\n",
    "    if(!exists('windowsRecomb')){\n",
    "        load('~/pophuman.uab.cat/Results/10kb/windowsRecomb.RData')\n",
    "    }  \n",
    "    \n",
    "    if(ecdf == FALSE & resampling == TRUE){\n",
    "        selectedColors <- colors[stat2colors %in% c(statistic,'globalRecomb')]\n",
    "       \n",
    "        temp <- paste0('~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/',statistic,'PvalueSig.RData')\n",
    "        file <- get(load(temp))\n",
    "        totalSampling <- length(file[[statistic]])\n",
    "        print(totalSampling)\n",
    "        \n",
    "        dens <- merge(file,windowsRecomb,by=c('chr','ID'))\n",
    "        dens <- melt(dens,measure.vars = 'recomb')\n",
    "        dens[['distr']] <- as.factor(paste0(statistic))\n",
    "        dens <- dens[,c('value','distr')]\n",
    "        \n",
    "        tempGlobalRecomb <- lapply(1:100,function(x)sample(windowsRecomb[['recomb']],totalSampling))\n",
    "        tempGlobalRecomb <- do.call(cbind,tempGlobalRecomb)\n",
    "        globalRecomb <- matrix(c(0),nrow=nrow(tempGlobalRecomb))\n",
    "        for(i in ncol(tempGlobalRecomb)){\n",
    "            temp <- tempGlobalRecomb[,i]\n",
    "            globalRecomb <- globalRecomb + temp\n",
    "        }\n",
    "        print(ncol(tempGlobalRecomb))\n",
    "        globalRecomb <- as.data.frame(matrix(globalRecomb/ncol(tempGlobalRecomb),ncol=1))\n",
    "        recombDens <- globalRecomb;colnames(recombDens) <- c('value')\n",
    "        recombDens[['distr']] <- as.factor(paste0('globalRecomb'))\n",
    "        print(head(recombDens))                 \n",
    "\n",
    "\n",
    "        densPlot <- rbind(recombDens,dens)\n",
    "        print(head(densPlot))\n",
    "        print(tail(densPlot))\n",
    "        limit <- quantile(na.omit(as.numeric(as.character(windowsRecomb[['recomb']]))),c(0.1,0.9))\n",
    "    \n",
    "        p <- ggplot(densPlot) + \n",
    "            geom_density(aes(x=value,fill=distr),alpha=0.3) +\n",
    "            scale_fill_manual(values = selectedColors) +\n",
    "            xlim(limit) +\n",
    "            labs(y='',x='Recomb',title='Cumulative Distribution',color='Statistics')+\n",
    "            theme(axis.text.x = element_text(size=32,face='bold',vjust = 0.5, angle = 90),legend.title=element_text(size=32), legend.text=element_text(size=30)) +\n",
    "            theme_bw()\n",
    "\n",
    "        return(p)                           \n",
    "                                   \n",
    "                                   \n",
    "    }                    \n",
    "    else if(ecdf == FALSE & resampling == FALSE){\n",
    "        selectedColors <- colors[stat2colors %in% c(statistic,'globalRecomb')]\n",
    "       \n",
    "        temp <- paste0('~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/',statistic,'PvalueSig.RData')\n",
    "        file <- get(load(temp))\n",
    "\n",
    "        totalSampling <- length(file[[statistic]])\n",
    "        \n",
    "        dens <- merge(file,windowsRecomb,by=c('chr','ID'))\n",
    "        \n",
    "        #recombDens <- melt(sample_n(windowsRecomb,totalSampling),measure.vars = 'recomb')\n",
    "        recombDens <- melt(windowsRecomb,measure.vars = 'recomb')\n",
    "        recombDens[['distr']] <- as.factor(paste0('globalRecomb'))\n",
    "\n",
    "        dens <- melt(dens,measure.vars = 'recomb')\n",
    "        dens <- dens[,c('ID','chr','startW','endW','variable','value')]\n",
    "        dens[['distr']] <- as.factor(paste0(statistic))\n",
    "\n",
    "        densPlot <- rbind(recombDens,dens)\n",
    "        limit <- quantile(na.omit(as.numeric(as.character(recombDens[['value']]))),c(0.1,0.9))\n",
    "\n",
    "        print(head(densPlot))\n",
    "        print(tail(densPlot))\n",
    "        \n",
    "        p <- ggplot(densPlot) + \n",
    "            geom_density(aes(x=value,fill=distr),alpha=0.3) +\n",
    "            scale_fill_manual(values = selectedColors) +\n",
    "            xlim(limit) +\n",
    "            labs(y='',x='Recomb',title='Cumulative Distribution',color='Statistics')+\n",
    "            theme(axis.text.x = element_text(size=32,face='bold',vjust = 0.5, angle = 90),legend.title=element_text(size=32), legend.text=element_text(size=30)) +\n",
    "            theme_bw()\n",
    "        \n",
    "        p2 <- ggplot(densPlot) + \n",
    "            geom_boxplot(aes(x=distr,y=value,fill=distr)) +\n",
    "            #scale_color_manual(values = colors) +\n",
    "            xlim(limit) +\n",
    "            labs(y='',x='Recomb',title='Cumulative Distribution',color='Statistics')+\n",
    "            theme(axis.text.x = element_text(size=32,face='bold',vjust = 0.5, angle = 90),legend.title=element_text(size=32), legend.text=element_text(size=30)) + \n",
    "            theme_bw()\n",
    "        \n",
    "        return(p)\n",
    "    }\n",
    "    else if (statistic == 'all'& ecdf == TRUE){\n",
    "        statistic <- c(\"Tajima_D\",\"FayWu_H\",\"iHS\",'Fst','XPEHH',\"FuLi_F\",\"FuLi_D\",'Pi','theta')\n",
    "        for(i in statistic){\n",
    "            temp <- paste0('~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/',i,'PvalueSig.RData')\n",
    "            file <- get(load(temp))\n",
    "            totalSampling <- length(file[[i]])\n",
    "                        \n",
    "            dens <- merge(file,windowsRecomb,by=c('chr','ID'))\n",
    "            \n",
    "            recombDens <- melt(sample_n(windowsRecomb,totalSampling),measure.vars = 'recomb')\n",
    "            recombDens[['distr']] <- as.factor(paste0('globalRecomb'))\n",
    "\n",
    "            dens <- melt(dens,measure.vars = 'recomb')\n",
    "            dens <- dens[,c('ID','chr','startW','endW','variable','value')]\n",
    "            dens[['distr']] <- as.factor(paste0(i,'Recomb'))\n",
    "            \n",
    "            if(i == 'Tajima_D'){\n",
    "                densPlot <- rbind(recombDens,dens)\n",
    "            }\n",
    "            else{\n",
    "                densPlot <- rbind(densPlot,dens)\n",
    "            }\n",
    "            \n",
    "        }\n",
    "         \n",
    "        \n",
    "        limit <- quantile(na.omit(as.numeric(as.character(windowsRecomb[['recomb']]))),c(0.1,0.9))\n",
    "        p <- ggplot(densPlot) + \n",
    "            stat_ecdf(aes(x=value,color=distr)) +\n",
    "            scale_color_manual(values = colors) +\n",
    "            xlim(limit) +\n",
    "            labs(y='',x='Recomb',title='Cumulative Distribution',color='Statistics')+\n",
    "            theme(axis.text.x = element_text(size=32,face='bold',vjust = 0.5, angle = 90),legend.title=element_text(size=32), legend.text=element_text(size=30)) + \n",
    "            theme_bw()\n",
    "    \n",
    "        \n",
    "        return(p)\n",
    "    }\n",
    "}\n",
    "\n",
    "stat <- c(\"Tajima_D\",\"FayWu_H\",\"iHS\",'Fst','XPEHH',\"FuLi_F\",\"FuLi_D\",'Pi','theta')\n",
    "\n",
    "distrRecombPlot(statistic='Tajima_D',ecdf = FALSE,resampling=F)                                   \n",
    "for(i in stat){\n",
    "    ggsave(filename = paste0('~/pophuman.uab.cat/Results/Plots/Features/Recomb/',i,'RecombDistr.png'),plot=distrRecombPlot(statistic=i,ecdf = FALSE,resampling=F),dpi = 150,height = 8,width = 6)\n",
    "}\n",
    "ggsave(filename = '~/pophuman.uab.cat/Results/Plots/Features/Recomb/cumulativeDistribution.png',plot=distrRecombPlot(statistic='all',ecdf = TRUE,resampling = F),dpi = 150,height = 10,width = 10)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-09T11:23:10.079189Z",
     "start_time": "2018-04-09T11:25:56.214Z"
    },
    "hidden": true
   },
   "source": [
    "#### Statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-28T10:29:37.259796Z",
     "start_time": "2018-05-28T10:33:16.429Z"
    },
    "code_folding": [
     0
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "distrRecombTable <- function(statistic){\n",
    "    recombTable <- data.frame()\n",
    "    if(!exists('windowsRecomb')){\n",
    "        load('~/pophuman.uab.cat/Results/10kb/windowsRecomb.RData')\n",
    "    }  \n",
    "    for(i in statistic){\n",
    "        print(i)\n",
    "        temp <- paste0('~/pophuman.uab.cat/Results/10kb/Pvalues/Significatives/',i,'PvalueSig.RData')\n",
    "        file <- get(load(temp))\n",
    "        totalSampling <- length(file[[i]])\n",
    "        \n",
    "        dens <- merge(file,windowsRecomb,by=c('chr','ID'))\n",
    "        wilcox <- wilcox.test(x = dens[['recomb']],sample(windowsRecomb[['recomb']],totalSampling))['p.value']\n",
    "        ks <- ks.test(x = dens[['recomb']],windowsRecomb[['recomb']])['p.value']\n",
    "        \n",
    "        temp <- as.data.frame(c(i,totalSampling,wilcox,ks));colnames(temp) <- c('statistic','samplingLength','Wilcoxon','Kolmogorov-Smirnov')\n",
    "    \n",
    "        recombTable <- rbind(recombTable,temp)\n",
    "    }\n",
    "    return(recombTable)\n",
    "}\n",
    "\n",
    "recombDistributionTable <- distrRecombTable(statistic=c(\"Tajima_D\",\"FayWu_H\",\"iHS\",'Fst','XPEHH',\"FuLi_F\",\"FuLi_D\",'Pi','theta'))\n",
    "fwrite(recombDistributionTable,file='~/pophuman.uab.cat/Results/10kb/Features/recombDistributionTable',sep='\\t',quote = F,row.names = F,col.names = T)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Ploting tiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-05-28T10:13:28.357455Z",
     "start_time": "2018-05-28T10:17:08.233Z"
    },
    "code_folding": [
     68,
     143
    ],
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "tileTop <- function(side, stat, topNumber){\n",
    "    file <- paste0('/data/shared/scan/Results/10kb/Pvalues/Significatives/',stat,'PvalueSig.RData')\n",
    "    data <- get(load(file))\n",
    "    rownames(data)<-NULL\n",
    "    data[['Pvalue']] <- as.numeric(as.character(data[['Pvalue']]))\n",
    "    data[[stat]] <- as.numeric(as.character(data[[stat]]))\n",
    "    # data <- data[data[['chr']]!='chrY',]\n",
    "    load('/data/shared/scan/Results/10kb/windowsCoord.RData')\n",
    "    load('/data/shared/scan/Results/10kb/genes2IDEnhancers.RData')\n",
    "    load('/data/shared/scan/Results/10kb/windowsRecomb.RData')\n",
    "    if(side == 'negative'){\n",
    "             # Extract only positive and top values by Group\n",
    "        if( stat == 'Pi' || stat == 'theta'){\n",
    "          data <- data[data[[stat]] != 0 & data[['chr']] != 'chrY' & data[['chr']] != 'chrX',]\n",
    "        }\n",
    "        subset <- data[data[[stat]] < 0,]\n",
    "        subset <- subset[order(subset[['Pvalue']], decreasing = F),]\n",
    "        subsetByGroup <- data[data[['Group']] %in% head(unique(subset[['Group']]), topNumber),]\n",
    "        subsetByGroup <- subsetByGroup[order(subsetByGroup[['Pvalue']]),]\n",
    "        windowsCoord[['ID']] <- as.numeric(windowsCoord[['ID']])\n",
    "        subsetByGroup[['ID']] <- as.numeric(as.character(subsetByGroup[['ID']]))\n",
    "        subsetByGroup <- merge(subsetByGroup, windowsCoord, by=c('chr','ID'))\n",
    "        #subsetByGroup <- merge(subsetByGroup, windowsRecomb, by=c('chr','ID','startW','endW'))\n",
    "\n",
    "        # Grouping IDs and create labels with minor and max starW and endW\n",
    "        #contigous2coord <- unique(subsetByGroup[,c('chr','pop',stat,'Pvalue','Group','recomb')])\n",
    "        contigous2coord <- unique(subsetByGroup[,c('chr','pop',stat,'Pvalue','Group')])\n",
    "        maxs <- aggregate(endW ~ Group, data = subsetByGroup, FUN = max)\n",
    "        mins <- aggregate(startW ~ Group, data = subsetByGroup, FUN = min)\n",
    "        coordsGroup <- merge(mins,maxs,by='Group')\n",
    "        #contigous2coord <- unique(contigous2coord[,c('chr','pop',stat,'Group','recomb')])\n",
    "        contigous2coord <- unique(contigous2coord[,c('chr','pop',stat,'Group')])\n",
    "        contigous2coord <- merge(contigous2coord, coordsGroup, by='Group')\n",
    "        #contigous2coord[['label']] <- paste0(contigous2coord[['chr']],':',contigous2coord[['startW']],'-',contigous2coord[['endW']],'|',round(as.numeric(as.character(contigous2coord[['recomb']])), digits = 3))\n",
    "        contigous2coord[['label']] <- paste0(contigous2coord[['chr']],':',contigous2coord[['startW']],'-',contigous2coord[['endW']])\n",
    "        contigous2coord <- unique(contigous2coord[,c('Group','label')])\n",
    "        subsetByGroup <- merge(subsetByGroup, contigous2coord,by='Group')\n",
    "\n",
    "        # Groups with minor Pvalue (highest stat signal)\n",
    "        groupPvalues <- unique(subsetByGroup[,c('chr','pop','Pvalue','Group')])\n",
    "        groupPvalues <- aggregate(Pvalue ~ Group+pop, data = groupPvalues, FUN = min)\n",
    "        # Groups with max stat value (highest stat signal)\n",
    "        groupStat <- unique(subsetByGroup[,c('Group','pop',stat)])\n",
    "        groupStat <- aggregate(groupStat[[stat]] ~ Group+pop, data = groupStat, FUN = max)\n",
    "        maxSignal <- merge(groupPvalues, groupStat,by=c('Group','pop'));colnames(maxSignal) <- c('Group','pop','Pvalue',stat) \n",
    "        #subsetByGroup <- subsetByGroup[,c('chr','ID','startW','endW','pop','Group','recomb','label')]\n",
    "        subsetByGroup <- subsetByGroup[,c('chr','ID','startW','endW','pop','Group','label')]\n",
    "\n",
    "        subsetByGroup <- merge(subsetByGroup, maxSignal, by = c('pop','Group'))\n",
    "\n",
    "        # Dcast in order to put genes name in geom_tile \n",
    "        subsetByGroup <- merge(subsetByGroup, genes2ID, by = 'ID')\n",
    "        # subsetByGroupCast <- dcast(subsetByGroup,Group+GeneID+ID+chr+startW+endW~pop, value.var = stat)\n",
    "        # subsetByGroupCast$GeneID[is.na(subsetByGroupCast$GeneID)]<-\"\"\n",
    "        # subsetByGroupCast <- unique(subsetByGroupCast[,c(1,2)])\n",
    "        subsetByGroupCast <- unique(subsetByGroup[,c('Group','GeneID')])\n",
    "    #    subsetByGroupCast <- ddply(subsetByGroupCast, .(Group), summarize,GeneID=toString(GeneID))\n",
    "        subsetByGroupCast <- subsetByGroupCast %>% group_by(Group) %>% summarize(GeneID=toString(GeneID))\n",
    "        subsetByGroupCast[['GeneID']][subsetByGroupCast[['GeneID']] == 'NA']<-\"\"\n",
    "        label <- unique(subsetByGroup[,c('Group','label')])\n",
    "        subsetByGroupCast <- merge(subsetByGroupCast,label,by='Group')\n",
    "        if(stat == 'XPEHH'){\n",
    "          p <- ggplot(data=subsetByGroup)+\n",
    "            geom_tile(aes(x=pop,y=label,fill=Pvalue),show.legend = T) + \n",
    "            xlab(\"Population\") + \n",
    "            ylab(\"Coordinates\")+\n",
    "            ggtitle(stat)+\n",
    "            theme(axis.text.x=element_text(vjust = 0.5, angle = 90),legend.position=\"left\")\n",
    "\n",
    "          pGenes <- ggplot(data=subsetByGroupCast)+\n",
    "            geom_blank(aes(x='a',y=label)) +\n",
    "            scale_y_discrete(labels = subsetByGroupCast$GeneID[order(subsetByGroupCast$label)])+  theme(axis.title = element_blank(),axis.text.x = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),axis.text.y=element_text(hjust = 0))\n",
    "\n",
    "          p <- plot_grid(p,pGenes,ncol = 2,rel_widths = c(1,1.5),align = \"h\")\n",
    "        }\n",
    "        else{\n",
    "          p <- ggplot(data=subsetByGroup)+\n",
    "            geom_tile(aes(x=pop,y=label,fill=Pvalue),show.legend = T) + \n",
    "            xlab(\"Population\") + \n",
    "            ylab(\"Coordinates\")+\n",
    "            ggtitle(stat)+\n",
    "            theme(axis.text.x=element_text(vjust = 0.5, angle = 90),legend.position=\"left\")\n",
    "\n",
    "          pGenes <- ggplot(data=subsetByGroupCast)+\n",
    "            geom_blank(aes(x='a',y=label)) +\n",
    "            scale_y_discrete(labels = subsetByGroupCast$GeneID[order(subsetByGroupCast$label)])+  theme(axis.title = element_blank(),axis.text.x = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),axis.text.y=element_text(hjust = 0))\n",
    "\n",
    "          p <- plot_grid(p,pGenes,ncol = 2,rel_widths = c(1,0.5),align = \"h\")\n",
    "        }\n",
    "\n",
    "  }\n",
    "    else if(side == 'positive'){\n",
    "        # Extract only positive and top values by Group\n",
    "        if( stat == 'Pi' || stat == 'theta'){\n",
    "          data <- data[data[[stat]] != 0 & data[['chr']] != 'chrY' & data[['chr']] != 'chrX',]\n",
    "        }\n",
    "        subset <- data[data[[stat]] > 0,]\n",
    "        subset <- subset[order(subset[['Pvalue']], decreasing = F),]\n",
    "        subsetByGroup <- data[data[['Group']] %in% head(unique(subset[['Group']]), topNumber),]\n",
    "        subsetByGroup <- subsetByGroup[order(subsetByGroup[['Pvalue']]),]\n",
    "        windowsCoord[['ID']] <- as.numeric(windowsCoord[['ID']])\n",
    "        subsetByGroup[['ID']] <- as.numeric(as.character(subsetByGroup[['ID']]))\n",
    "        subsetByGroup <- merge(subsetByGroup, windowsCoord, by=c('chr','ID'))\n",
    "        #subsetByGroup <- merge(subsetByGroup, windowsRecomb, by=c('chr','ID','startW','endW'))\n",
    "\n",
    "        # Grouping IDs and create labels with minor and max starW and endW\n",
    "        #contigous2coord <- unique(subsetByGroup[,c('chr','pop',stat,'Pvalue','Group','recomb')])\n",
    "        contigous2coord <- unique(subsetByGroup[,c('chr','pop',stat,'Pvalue','Group')])\n",
    "        maxs <- aggregate(endW ~ Group, data = subsetByGroup, FUN = max)\n",
    "        mins <- aggregate(startW ~ Group, data = subsetByGroup, FUN = min)\n",
    "        coordsGroup <- merge(mins,maxs,by='Group')\n",
    "        #contigous2coord <- unique(contigous2coord[,c('chr','pop',stat,'Group','recomb')])\n",
    "        contigous2coord <- unique(contigous2coord[,c('chr','pop',stat,'Group')])\n",
    "        contigous2coord <- merge(contigous2coord, coordsGroup, by='Group')\n",
    "        #contigous2coord[['label']] <- paste0(contigous2coord[['chr']],':',contigous2coord[['startW']],'-',contigous2coord[['endW']],'|',round(as.numeric(as.character(contigous2coord[['recomb']])), digits = 3))\n",
    "        contigous2coord[['label']] <- paste0(contigous2coord[['chr']],':',contigous2coord[['startW']],'-',contigous2coord[['endW']])\n",
    "        contigous2coord <- unique(contigous2coord[,c('Group','label')])\n",
    "        subsetByGroup <- merge(subsetByGroup, contigous2coord,by='Group')\n",
    "\n",
    "        # Groups with minor Pvalue (highest stat signal)\n",
    "        groupPvalues <- unique(subsetByGroup[,c('chr','pop','Pvalue','Group')])\n",
    "        groupPvalues <- aggregate(Pvalue ~ Group+pop, data = groupPvalues, FUN = min)\n",
    "        # Groups with max stat value (highest stat signal)\n",
    "        groupStat <- unique(subsetByGroup[,c('Group','pop',stat)])\n",
    "        groupStat <- aggregate(groupStat[[stat]] ~ Group+pop, data = groupStat, FUN = max)\n",
    "        maxSignal <- merge(groupPvalues, groupStat,by=c('Group','pop'));colnames(maxSignal) <- c('Group','pop','Pvalue',stat) \n",
    "        #subsetByGroup <- subsetByGroup[,c('chr','ID','startW','endW','pop','Group','recomb','label')]\n",
    "        subsetByGroup <- subsetByGroup[,c('chr','ID','startW','endW','pop','Group','label')]\n",
    "\n",
    "        subsetByGroup <- merge(subsetByGroup, maxSignal, by = c('pop','Group'))\n",
    "\n",
    "        # Dcast in order to put genes name in geom_tile \n",
    "        subsetByGroup <- merge(subsetByGroup, genes2ID, by = 'ID')\n",
    "        # subsetByGroupCast <- dcast(subsetByGroup,Group+GeneID+ID+chr+startW+endW~pop, value.var = stat)\n",
    "        # subsetByGroupCast$GeneID[is.na(subsetByGroupCast$GeneID)]<-\"\"\n",
    "        # subsetByGroupCast <- unique(subsetByGroupCast[,c(1,2)])\n",
    "        subsetByGroupCast <- unique(subsetByGroup[,c('Group','GeneID')])\n",
    "    #    subsetByGroupCast <- ddply(subsetByGroupCast, .(Group), summarize,GeneID=toString(GeneID))\n",
    "        subsetByGroupCast <- subsetByGroupCast %>% group_by(Group) %>% summarize(GeneID=toString(GeneID))\n",
    "        subsetByGroupCast[['GeneID']][subsetByGroupCast[['GeneID']] == 'NA']<-\"\"\n",
    "        label <- unique(subsetByGroup[,c('Group','label')])\n",
    "        subsetByGroupCast <- merge(subsetByGroupCast,label,by='Group')\n",
    "        if(stat == 'XPEHH'){\n",
    "          p <- ggplot(data=subsetByGroup)+\n",
    "            geom_tile(aes(x=pop,y=label,fill=Pvalue),show.legend = T) + \n",
    "            xlab(\"Population\") + \n",
    "            ylab(\"Coordinates\")+\n",
    "            ggtitle(stat)+\n",
    "            theme(axis.text.x=element_text(vjust = 0.5, angle = 90),legend.position=\"left\")\n",
    "\n",
    "          pGenes <- ggplot(data=subsetByGroupCast)+\n",
    "            geom_blank(aes(x='a',y=label)) +\n",
    "            scale_y_discrete(labels = subsetByGroupCast$GeneID[order(subsetByGroupCast$label)])+  theme(axis.title = element_blank(),axis.text.x = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),axis.text.y=element_text(hjust = 0))\n",
    "\n",
    "          p <- plot_grid(p,pGenes,ncol = 2,rel_widths = c(1,1.5),align = \"h\")\n",
    "        }\n",
    "        else{\n",
    "          p <- ggplot(data=subsetByGroup)+\n",
    "            geom_tile(aes(x=pop,y=label,fill=Pvalue),show.legend = T) + \n",
    "            xlab(\"Population\") + \n",
    "            ylab(\"Coordinates\")+\n",
    "            ggtitle(stat)+\n",
    "            theme(axis.text.x=element_text(vjust = 0.5, angle = 90),legend.position=\"left\")\n",
    "\n",
    "          pGenes <- ggplot(data=subsetByGroupCast)+\n",
    "            geom_blank(aes(x='a',y=label)) +\n",
    "            scale_y_discrete(labels = subsetByGroupCast$GeneID[order(subsetByGroupCast$label)])+  theme(axis.title = element_blank(),axis.text.x = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),axis.text.y=element_text(hjust = 0))\n",
    "\n",
    "          p <- plot_grid(p,pGenes,ncol = 2,rel_widths = c(1,0.5),align = \"h\")\n",
    "        }\n",
    "  }\n",
    "  return(p)\n",
    "}"
   ]
  }
 ],
 "metadata": {
  "hide_input": false,
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.3.2"
  },
  "toc": {
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "165px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "position": {
    "height": "145px",
    "left": "914px",
    "right": "16px",
    "top": "119px",
    "width": "350px"
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
